# **CALENDARIOS DEPORTIVOS**

## Carga de datos

In [ ]:
from google.colab import files #importar modulo
uploaded = files.upload()

Saving Calendario_Deportivo.csv to Calendario_Deportivo.csv


In [ ]:
import plotly.express as px


**Calendario Deportivo**

Subimos la base de datos desde el Local

In [ ]:
#Importamos los paquetes
import pandas as pd
import io
#Se leyo la base
df=pd.read_csv(io.StringIO(uploaded['Calendario_Deportivo.csv'].decode('utf-8')),sep=';')
df.head(5)
#pd.set_option("display.max_rows", None)

NOMBRE                   DEPORTE                      TIPO  \
0        VELA  CALENDARIO INTERNACIONAL  CALENDARIO INTERNACIONAL   
1        VELA  CALENDARIO INTERNACIONAL  CALENDARIO INTERNACIONAL   
2  BALONCESTO       CALENDARIO NACIONAL       CALENDARIO NACIONAL   
3        TEJO       CALENDARIO NACIONAL       CALENDARIO NACIONAL   
4       LUCHA       CALENDARIO NACIONAL       CALENDARIO NACIONAL   

                    FECHA      PAIS                    LUGAR Unnamed: 6  \
0  12/26/2018 12:00:00 AM       USA    MIAMI- ESTADOS UNIDOS        NaN   
1  12/26/2018 12:00:00 AM     Chile             PUCÓN CHILE        NaN   
2  12/18/2018 12:00:00 AM  Colombia  TUNJA (BOYACA) COLOMBIA        NaN   
3  12/14/2018 12:00:00 AM  Colombia    GIRARDOT CUNDINAMARCA        NaN   
4  12/13/2018 12:00:00 AM  Colombia                  NECOCLI        NaN   

  Unnamed: 7  
0        NaN  
1        NaN  
2        NaN  
3        NaN  
4        NaN

In [ ]:
#importar paquetes
import pandas as pd
import numpy as np

In [ ]:
#importamos paquete de fechas
import datetime
from datetime import datetime
df['FECHA']= pd.to_datetime(df['FECHA'], dayfirst=True, errors='coerce')# Convertir la fecha en el formato correcto
#Para cambiar el formato de la fecha tuve que agg dayfirst=True y errors='coerce'
type(df['FECHA'][4])

pandas._libs.tslibs.timestamps.Timestamp

## Depuracion


In [ ]:
# Borrar columnas, sin datos
df.drop(['Unnamed: 6','Unnamed: 7','DEPORTE'], axis=1, inplace=True)

In [ ]:
df=df.rename(columns={'TIPO':'TIPO DE CALENDARIO'})

In [ ]:
# Renombrar una columna
df=df.rename(columns={'NOMBRE':'DEPORTE'})

In [ ]:
df['DEPORTE']=df['DEPORTE'].replace({'TIRO TODAS':'TIRO DEPORTIVO','TIRO ESCOPETAS':'TIRO DEPORTIVO','TIRO TRAP AUTOMATICO':'TIRO DEPORTIVO','TIRO RIFLES':'TIRO DEPORTIVO','TIRO PISTOLAS':'TIRO DEPORTIVO','TIRO FOSA UNIVERSAL':'TIRO DEPORTIVO','TIRO RIFLES Y PISTOLAS':'TIRO DEPORTIVO','TIRO FOSA OLIMPICA':'TIRO DEPORTIVO','TIRO SPORTING CLAYS':'TIRO DEPORTIVO','TIRO':'TIRO DEPORTIVO','DEPORTES AÉREOS':'DEPORTES AEREOS','TIRO DOBLE TRAP Y FOSA UNIVERSAL':'TIRO DEPORTIVO','TIRO COMPAC SPORTING':'TIRO DEPORTIVO'})

In [ ]:
df['TIPO DE CALENDARIO']=df['TIPO DE CALENDARIO'].replace({'CAMPEONATO BOLIVARIANO':'CALENDARIO INTERNACIONAL','CALENDARIO INTERNACINAL':'CALENDARIO INTERNACIONAL','OTRO INTERNACIONAL':'CALENDARIO INTERNACIONAL','NACIONAL':'CALENDARIO NACIONAL','INTERNACIONAL':'CALENDARIO INTERNACIONAL','CAMPEONATO MUNDIAL':'CALENDARIO INTERNACIONAL','CAMPEONATO SURAMERICANO':'CALENDARIO INTERNACIONAL','ACADÉMICO':'ACADEMICO','CAMPEONATO PANAMERICANO':'CALENDARIO INTERNACIONAL','OPEN INTERNACIONAL':'CALENDARIO INTERNACIONAL','CONCENTRACIÓN':'CONCENTRACION','GRAN PRIX INTERNACIONAL':'CALENDARIO INTERNACIONAL',' SPORTING CLAYS':'CALENDARIO NACIONAL','SHOWDANCE BAILE DEPORTIVO':'CALENDARIO NACIONAL','SALSA SHOWDANCE':'CALENDARIO NACIONAL','JUEGOS NACIONALES':'CALENDARIO NACIONAL','SUDAMERICANO':'CALENDARIO INTERNACIONAL','CAMPEONATO IBEROAMERICANO':'CALENDARIO INTERNACIONAL','ENCUENTRO NACIONAL DE FOLCLORE DEPORTIVO':'CALENDARIO NACIONAL','CAPACITACIÓN':'CAPACITACION','V FORMACION DE JUECES Y ENTRENADORES':'CAPACITACION','CAMPAMENTO':'CALENDARIO NACIONAL','BOGOTA':'CALENDARIO NACIONAL','OPEN DEL CAFÉ':'CALENDARIO NACIONAL','OPEN GERMANY':'CALENDARIO INTERNACIONAL','PRECLASIFICATORIO':'CALENDARIO NACIONAL','II CAMPEONATO PANAMERICANO DE BAILE DEPORTIVO':'CALENDARIO INTERNACIONAL','ENTRENADORES':'CAPACITACION','CALENDARIO INTERNACIONAL.':'CALENDARIO INTERNACIONAL','MUNDIAL':'CALENDARIO INTERNACIONAL','RESERVA DEPORTIVA':'CALENDARIO NACIONAL','BRIDGE':'CALENDARIO NACIONAL','CAMPEONATO LATINOAMERICANO':'CALENDARIO INTERNACIONAL','MARATON':'CALENDARIO INTERNACIONAL','CLASIFICATORIO':'CALENDARIO INTERNACIONAL','CAMPEONATO CENTROAMERICANO':'CALENDARIO INTERNACIONAL','TECNICO-METODOLOGICO':'CAPACITACION','EVENTO DEL CICLO OLÍMPICO':'CALENDARIO INTERNACIONAL','COPA MUNDO':'CALENDARIO INTERNACIONAL','CALENDARIO UNICO':'CALENDARIO INTERNACIONAL','COMPETENCIA':'CALENDARIO INTERNACIONAL'})

In [ ]:
# Definir una lista con las condiciones
lista = df[df['TIPO DE CALENDARIO'] == 'POVENZA'].index
lista1 = df[df['TIPO DE CALENDARIO'] == 'WOOPERTTAL'].index
listipo=[lista[0],lista1[0]]

In [ ]:
df=df.drop(listipo)

In [ ]:
df['PAIS'] = df['PAIS'].replace({'USA':'EEUU','EE.UU':'EEUU','ESPAñA':'ESPAÑA','PERÊ':'OTROS','BOYACA':'COLOMBIA','TOLIMA':'COLOMBIA','SANTA MARTA':'COLOMBIA','PAIPA,':'COLOMBIA','POR  DEFINIR':'POR DEFINIR','SIN DEFINIR':'POR DEFINIR','CALI':'COLOMBIA','ESTADOS UNIDOS':'EEUU','BOGOTA':'COLOMBIA','ESPAÑA':'ESPAÑA','VALLE':'COLOMBIA','COLOMBA':'COLOMBIA','META':'COLOMBIA','LIMA':'PERU','SANTIAGO':'CHILE','MÉXICO':'MEXICO','A CONFIRMAR':'POR DEFINIR','MEDELLIN':'COLOMBIA','BOLIVAR':'COLOMBIA','PARIS':'FRANCIA','MONTEMOR':'PORTUGAL','EUROPA':'OTROS','VALLE DEL CAUCA':'COLOMBIA','ANTIOQUIA':'COLOMBIA','ROLDANILLO':'COLOMBIA','CIUDAD DE MEXICO':'MEXICO','MADRID':'ESPAÑA','SANTIAGO DE CHILE':'CHILE','BOGOTÁ D. C.':'COLOMBIA','CÊCUTA':'OTROS','COLOMABIA':'COLOMBIA','ARACAJU':'BRASIL','REPUBLICA DE CHINA':'CHINA','IBAGUE (TOLIMA)':'COLOMBIA','CORDOBA':'COLOMBIA','CUBARRAL,':'COLOMBIA','UNITED ARAB EMIRATES':'EMIRATOS ARABAES','POR DEFIIR':'POR DEFINIR','TODAS':'OTROS','CASANARE':'COLOMBIA','BUENOS AIRES':'ARGENTINA','CASTRIES, STA. LUCIA':'SANTA LUCIA','SANTO DOMINGO':'REPUBLICA DOMINICANA','RUSSIA':'RUSIA','REP. DOMINICANA':'REPUBLICA DOMINICANA','REP DOMINICANA':'REPUBLICA DOMINICANA','PAIPA':'COLOMBIA','SANTANDER':'COLOMBIA','BIOLURUSSIA':'BIELURUSIA','GUAYAQUIL, ECUADOR':'ECUADOR','LONATO - ITALIA':'ITALIA','SALVADOR':'EL SALVADOR','BIELURUSSIA':'BIELORUSIA','BOGOTA D.C.':'COLOMBIA','BOG-MIL':'COLOMBIA','EE UU':'EEUU','EUROPA Y ASIA':'OTROS','P. RICO Y CANADÁ':'PUERTO RICO','TOCANCIPA':'COLOMBIA','MALASYA':'MALASIA','SINCELEJO (SUCRE)':'COLOMBIA','BARCELONA':'ESPAÑA','BOY-CAS':'COLOMBIA','EMIRATOS ARABES UNIDOS':'EMIRATOS ARABES','VENDEE':'FRANCIA','MEX-CAN':'MEXICO','PRADO  TOLIMA':'COLOMBIA','HOMSTED MIAMI':'EEUU','ROMA':'ITALIA','CHN-UAE-RUS-GBR-CAN-MEX':'CHINA','CHINA POPULAR':'CHINA','WASQUEHAL':'FRANCIA','NETHERLANDS':'HOLANDA','SWE':'SUIZA','MODENA':'ITALIA','HUI':'COLOMBIA','VARIOS':'OTROS','BELARUS':'BIELORUSIA','SOFIA':'BULGARIA','GUATAPE - ANTIOQUIA':'COLOMBIA','MAGDALENA':'COLOMBIA','CHOCO':'COLOMBIA','CANADÃ':'CANADA','P.D':'POR DEFINIR','FRANCIA  ESPAñA  ITALIA':'FRANCIA','GREAT BRITAIN':'GRAN BRETAÑA','OKLAHOMA CITY':'EEUU','ATL-BOL-COR-MAG':'COLOMBIA','POR ASIGNAR':'POR DEFINIR','GHI':'BOSNIA','SINGAPURE':'SINGAPUR','SUCRE':'COLOMBIA','TUNICIA':'TUNEZ','AZERBAIJAN':'AZERBAYAN','R. DOMINICANA':'REPUBLICA DOMINICANA','NACIONAL':'OTROS','TORONTO':'CANADA','SURREY  BC':'CANADA','GWANGJU':'KOREA','BRASIL-CUBA':'BRASIL','ANT':'COLOMBIA','SAN-NOR':'COLOMBIA','CAU-NAR':'COLOMBIA','GERMANIA':'ALEMANIA','SAN GIL  SANTANDER':'COLOMBIA','UK':'REINO UNIDO','IBAGUE':'COLOMBIA','HCMC-HOCHIMINH CITY':'OTROS','BRASIL - CUBA':'BRASIL','GRAND BRETAñA':'REINO UNIDO','VAL':'OTROS','AMA':'OTROS','UZBEKSITAN':'UZBEKISTAN','CAZHAJTAN':'KAZAJSTAN','BOSNIA AND HERZEGOVINA':'BOSNIA','CARGENTINA':'ARGENTINA','SANCTI ESPÍRITUS':'OTROS','SANCTI ESPÍRITUS':'OTROS','AZERBAYJAN':'AZERBAYAN','ROMANIA':'RUMANIA','CIENAGA':'COLOMBIA','COREO':'KOREA','HAMBURG, ALEMANIA':'ALEMANIA','RACICE':'REPUBLICA CHECA','MEDELLÍN':'COLOMBIA','TERVUREN':'BELGICA','DEPORTIVO':'SIN DEFINIR','GRAND BRETAÑA':'REINO UNIDO','TEMPERE':'FINLANDIA','TORONTO, CANADA':'CANADA','BUCAREST':'RUMANIA','MOSCU, TOKIO':'RUSIA','BERLIN':'ALEMANIA','PEREIRA':'COLOMBIA','GUADALAJARA':'MEXICO','MONTEVELHO,':'PORTUGAL','HAVANA, CUBA':'CUBA','ROMA, ITALIA':'ITALIA','ATHENAS':'GRECIA','SAN JOSE, COSTA RICA':'COSTA RICA','MANAGUA, NICARAGUA':'NICARAGUA','MARRAKECH':'MARRUECOS','SANTA MARTA, COLOMBIA':'COLOMBIA','NEW YORK':'EEUU','GIRARDOT - C/MARCA':'COLOMBIA','ZAMORA, ECUADOR':'ECUADOR','SACATEPEQUEZ, GUATEMALA':'GUATEMALA','COMPETENCIA':'OTROS','CUMANA, VENEZUELA':'VENEZUELA','VARGAS':'ESPAÑA','U-11, U-13':'OTROS','CASTELLETTO - ITALIA':'ITALIA','LA ROMANA, REPÊBLICA DOMINICANA':'REPUBLICA DOMINICANA','GUATAPÉ':'COLOMBIA','SANTIAGO DE CHILE, CHILE':'CHILE','CARTAGENA':'COLOMBIA','LOULE':'PORTUGAL','MONTREAL':'CANADA','SZOMBATHELY':'HUNGRIA','ALMEIDA':'PORTUGAL','GUAYAQUIL':'ECUADOR','BOGOTÁ':'COLOMBIA','SANTIAGO, REPÊBLICA DOMINICANA':'REPUBLICA DOMINICANA','EMIRATOS ÁRABES':'EMIRATOS ARABES','ATLANTICO':'COLOMBIA','ESTADO UNIDOS':'EEUU','KAMPALA':'UGANDA','VILLAVICENCIO - META':'COLOMBIA','MONACO':'FRANCIA','MIAMI - EEUU':'EUU','CAMAGUEY':'CUBA','GEORGETOWN, GUYANA':'GUYANA','PANAMÁ':'PANAMA','AMERICA':'OTROS','CANADÁ':'CANADA','WARLOK':'OTROS','EUROPA / AMERICA':'OTROS','SAN JOSE':'COSTA RICA','SALVADOR - GUATEMALA':'GUATEMALA','PD':'OTROS','NEIVA':'COLOMBIA','U-15, U-21':'OTROS','VALENCIA, VENEZUELA':'VENEZUELA','RISARALDA':'COLOMBIA','TOCANCIPA - COLOMBIA':'COLOMBIA','JAPÓN':'JAPON','LA HABANA, CUBA':'CUBA','GUATAVITA':'COLOMBIA','CIUDAD DE GUATEMALA':'GUATEMALA','SOGAMOSO':'COLOMBIA','BELGRADE':'SERBIA','SASSARI':'ITALIA','SZEGUED,':'HUNGRIA','IQUIQUE, CHILE':'CHILE','OSIJEK':'CROACIA','DORMAGEN':'ALEMANIA','KOPER':'ESLOVENIA','CAGUAS, PUERTO RICO':'PUERTO RICO','VILLAVICENCIO':'COLOMBIA','U.S.A.':'EEUU','CAJICA - COLOMBIA':'COLOMBIA','EMIRATOS ARBAES':'EMIRATOS ARABAES','CANADÃ':'CANADA','KAZAJSTAN':'KAZAJISTAN','TOL-CUN':'COLOMBIA','GRAN BRETAÑA':'REINO UNIDO','BHI':'OTROS','EUU':'EEUU','PASTO - NARIÑO':'COLOMBIA','SIN DEFINIR':'POR DEFINIR','PITETI':'OTROS','CAJICA - COLOMBIA':'COLOMBIA','EL SAVADOR':'EL SALVADOR','AZERBAIYÁN':'AZERBAIYAN','OBERHAUSEN':'ALEMANIA','PASTO':'COLOMBIA','RIS-CLA-QUI':'COLOMBIA','AZERBAYAN':'AZERBAIYAN','CANADÃ':'CANADA','EMIRATOS ARABAES':'EMIRATOS ARABES','INDONEDIA':'INDONESIA','CANADÃ\xa0':'CANADA'})

In [ ]:
#poner todo en mayuscula
df['PAIS'] = df['PAIS'].str.upper()

In [ ]:
lista = df[df['PAIS'] == '3/03/2017 0:00'].index
lista1 = df[df['PAIS'] == '10/05/2017 0:00'].index
lista2 = df[df['PAIS'] == '05/19/2017 12:00:00 AM'].index
lista3 = df[df['PAIS'] == '05/19/2017 12:00:00 AM'].index
lista4 = df[df['PAIS'] == 'MAYORES'].index
lista5 = df[df['PAIS'] == 'JUVENIL'].index
lista6 = df[df['PAIS'] == 'INFANTIL'].index
lista7 = df[df['PAIS'] == '08/18/2017 12:00:00 AM'].index
lista8 = df[df['PAIS'] == '07/20/2017 12:00:00 AM'].index
lista9 = df[df['PAIS'] == 'PRE INFANTIL'].index
lista10 = df[df['PAIS'] == 'U13 U15 U18 MAYORES'].index
lispais=[lista[0],lista1[0],lista2[0],lista3[0],lista4[0],lista4[0],lista5[0],lista6[0],lista7[0],lista8[0],lista9[0],lista10[0]]

In [ ]:
df=df.drop(lispais)

In [ ]:
df['LUGAR'] = df['LUGAR'].replace({'CARACAS Y VALENCIA':'CARACAS-VALENCIA','LONG ISLAND  NY':'LONG ISLAND','TAME (ARAUCA)':'TAME','ACACIAS (META)':'ACACIAS','CALI €“ VALLE DEL CAUCA':'CALI','OROCUE (CASANARE)':'OROCUE','CUENCA / ECUADOR':'CUENCA','SANTA ROSALIA (VICHADA)':'SANTA ROSALIA','13 - 15 FUSAGASUGA':'FUSAGASUGA','MEX-CAN':'MEXICO-CANADA','GUARNE (ANT)':'GUARNE','CHN-UAE-RUS-GBR-CAN-MEX':'CHINA-EMIRATOS ARABES-REINO UNIDO-CANADA-MEXICO','DENVER CO':'DENVER','BOGOTA Y BONSA':'BOGOTA-BONSA','AGUAZUL (CASANARE)':'AGUAZUL','SAO_PAULO':'SAO PAULO','GREEN BAY  WISCONSIN':'GREEN BAY','TRINIDAD(CASANARE)':'TRINIDAD','BOGOTA- BONZA':'BOGOTA-BONZA','LA CEJA - ANT.':'LA CEJA','RIONEGRO (ANT)':'RIONEGRO','VIRTUAL/ FECODAZ':'VIRTUAL-FECODAZ','TORONTO (CAN)':'TORONTO','TORONTO/CANADA':'TORONTO','SAN DIEGO  CA':'SAN DIEGO','IBAGUE - TOLIMA':'IBAGUE','BARRANQUILLA-COLOMBIA':'BARRANQUILLA','PUN TA CANA':'PUNTA CANA','PEREIRA - RISARALDA':'PEREIRA','BRIENZA_AXALP':'BRIENZA AXALP','BOGOTA/CUNDINAM':'BOGOTA','MONTERREY (CASANARE)':'MONTERREY','MARINIÑÑA':'MARINILLA','SAN GIL - SAN':'SAN GIL','RIO - HABANA':'RIO-HABANA','GUATAPE ANTIOQUIA':'GUATAPE','SAN ANTONIO (TEXAS)':'SAN ANTONIO','PORE (CASANARE)':'PORE','LIBANO - TOLIMA':'LIBANO','QUIBDO - CHOCO':'QUIBDO','BOJACA CUNDINAMARCA':'BOJACA','PALESTINA - CALDAS':'PALESTINA','BOGOTA - CALI':'BOGOTA-CALI','RIONEGRO - ANT.':'RIONEGRO','PALESTINA - CALDAS':'PALESTINA','YOPAL (CASANARE)':'YOPAL','BOGOTÃ\x81  (CLUB LOS ARRAYANES':'BOGOTA','CARTAGENA. BOLÃ\x8dVAR':'CARTAGENA','MIAMI, FL':'MIAMI','MIAMI, FLORIDA':'MIAMI','OCALA - FLORIDA':'OCALA-FLORIDA','SÃƑO JOSÉ DOS CAMPOS':'SAN JOSE DOS CAMPOS','BOGOTÃ\x81  (CLUB HATOGRANDE)':'BOGOTA','SAN JUAN Y GATINEAU':'SAN JUAN-GATINEAU','EUROPA -ASIA':'EUROPA-ASIA','FRISCO - TEXAS':'FRISCO-TEXAS','PALM BEACH - FLORIDA':'PALM BEACH','LIGNANO - SABBIADORO':'LIGNANO-SABBIADORO','MANIZALES, CALDAS':'MANIZALES','CARTAGENA-BOLIVAR':'CARTAGENA','GUATAPÉ':'GUATAPE','CARTAGENA-BOLIVAR':'CARTAGENA','STO DOMINGO':'SANTO DOMINGO','BOGOTÃ\x81, CUNDINAMARCA':'BOGOTA','CULIACAN SINALOA':'CULIACAN','PEREIRA-RISARALDA':'PEREIRA','SAN JUAN PTO RICO':'SAN JUAN','COTA CUNDINAMARCA PUEBLO VIEJO':'COTA-PUEBLO VIEJO','FECODAZ/BOGOTÃ\x81':'FECODAZ-BOGOTA','VILLA DE LEYVA CUND':'VILLA DE LEYVA','MISSANO Y TRIETE':'MISSANO-TRIETE','ASIAGO - ROANA':'ASIAGO-ROANA','BARRANQUILA-ATLANTICO':'BARRANQUILA','BOGOTÃ\x81, MEDELLIN, CALI':'BOGOTA-MEDELLIN-CALI','BAKÊ':'BAKAS','NEIVA Y TUNJA':'NEIVA-TUNJA','IBAGUÉ, TOLIMA':'IBAGUE','BUCARAMANGA-SANTANDER':'BUCARAMANGA','B/MANGA':'BUCARAMANGA','MEDELLÃ\x8dN, ANTIOQUIA':'MEDELLIN','LUBBOCK, TEXAS':'LUBBOCK','RESTREPO - VAL.':'RESTREPO','RÃ\x8dO DE JANEIRO':'RIO DE JANEIRO','RIO NEGRO ANT.':'RIO NEGRO','UBATE CUND':'UBATE','POPAYÃ\x81N':'POPAYAN','CUND':'CUNDINAMARCA','TAOYUAN, CHINESE TAIPEI':'TAOYUAN-TAIPEI','ATLÃ\x81NTICO':'ATLANTICO','3 ZONAS DEL PAIS':'COLOMBIA','BOGOTA BACATA':'BOGOTA-BACATA','SAN ANDRES ISLAS':'SAN ANDRES','DUITAMA BONZA':'DUITAMA-BONZA','PUERTO TRIUNFO  A CARTAGENA':'PUERTO TRIUNFO-CARTAGENA','FRANCIA/ARGENTINA /BRASIL/CUBA/ESPAÑA/ HUNGRIA/ITALIA/POLONIA':'FRANCIA-ARGENTINA-BRASIL-CUBA-ESPAÑA-HUNGRIA-ITALIA-POLONIA','FRANCIA/ITALIA/ESPAÑA/ALEMANIA':'FRANCIA-ITALIA-ESPAÑA-ALEMANIA','8 CIUDADES':'VARIOS','SUBACHOQUE CUND':'SUBACHOQUE','XI€™AN':'OTROS','ANAPOLIS, MAYLAND':'ANAPOLIS-MAYLAND','P/D':'POR DEFINIR','BUCARMANGA. SANTANDER':'BUCARMANGA','KAZÃ\x81N':'OTROS','BARRANQUILLA - CARTAGENA':'BARRANQUILLA-CARTAGENA','TBA':'OTROS','SIN DEFINIR':'POR DEFINIR','CAVASA - CALI':'CAVASA-CALI','LAGUNA SECA - CHIA':'LAGUNA SECA-CHIA','CAJICA LA HA':'CAJICA','CALI CCC':'CALI','BOGOTA, MANIZALEZ, MEDELLIN, BARRANQUILLA, CALI':'BOGOTA-MANIZALEZ-MEDELLIN-BARRANQUILLA-CALI','PRADO - TOLIMA':'PRADO','MEDELLIN-ANTIOQUIA':'MEDELLIN','ARJONA-BOLIVAR':'ARJONA','B/QUILLA':'BARRANQUILLA','BOGOTA EEEE':'BOGOTA','BOGOTA RANCHO':'BOGOTA','POR ASIGNAR':'POR DEFINIR','DESCENTRALIZADO':'VARIOS','CIUDAD DE PANAMÃ\x81':'CIUDAD DE PANAMA','RÃ\x8dO GRANDE':'OTROS','VALLE DE BRAVO - MEXICO':'VALLE DE BRAVO','VIÑA DEL MAR':'VIA DEL MAR','SANTIAGO, REPÊBLICA DOMINICANA':'SANTIAGO-REPUBLICA DOMINICANA','AUTODROMO DE TOCANCIPA':'TOCANCIPA','JERICO - ANTIOQUIA':'JERICO','VARIAS CIUDADES':'VARIOS','BARRANQUILLA - ATLANTICO':'BARRANQUILLA','EUROPA':'VARIOS','GEORGETOWN, GUYANA':'GEORGETOWN','PUERTO LOPEZ ( META)':'PUERTO LOPEZ','RUITOQUE - SANTANDER':'RUITOQUE','QUERÉTARO':'QUERETARO','FLANDES -  TOLIMA':'FLANDES','COEUR DE SAVOIE - FRANCIA':'COEUR DE SAVOIE','LIGA DE SQUASH DE VALLE':'VALLE DEL CAUCA','BOGOTA, CALI, MEDELLIN':'BOGOTA-CALI-MEDELLIN','IQUIQUE, CHILE':'IQUIQUE','HUNGRÃ\x8dA':'HUNGRIA','A DEFINIR':'POR DEFINIR','BOGOTÃ\x81 D. C.':'BOGOTA','A CONFIRMAR':'POR DEFINIR','IBAGUÉ':'IBAGUE','LA HABANA, CUBA':'LA HABANA','LIGA DE SQUASH DE ANTIOQUIA-COUNTRY EJECUTIVOS':'ANTIOQUIA','IBAGUE TOLIMA':'IBAGUE','CARTAGENA DE INDIAS':'CARTAGENA','PROVIDENCIA ISLAS':'PROVIDENCIA','APIA - RISARALDA':'APIA','CALI/CUCUTA/PEREIRA/MEDELLIN/BOGOTA/BARRANQUILLA/RIOACHA':'CALI-CUCUTA-PEREIRA-MEDELLIN-BOGOTA-BARRANQUILLA-RIOACHA','LAGO CALIMA - VALLE':'LAGO CALIMA','MOLVENO - ITALIA':'MOLVENO','LIGA DE SQUASH DE META':'META','VALENCIA, VENEZUELA':'VALENCIA','IBAGUÉ TOLIMA':'IBAGUE','COSTA NORTE, MEDELLIN,CALI':'COSTA NORTE-MEDELLIN-CALI','ESPAÑA':'ESPAÑA','RUSIA, ASIA':'RUSIA-ASIA','COSTA NORTE, ZONA URABA. MEDELLIN, CALI':'COSTA NORTE-URABA-MEDELLIN-CALI','ASUNCIÓN':'ASUNCION','PALM BEACH, FLORIDA':'PALM BEACH','AGUAZUL - CASANARE':'AGUAZUL-CASANARE','TORONTO, CANADA':'TORONTO','LIGA DE SQUASH DE ANTIOQUIA- RODEO':'EL RODEO','PERÊ':'OTROS','RIONEGRO ANT':'RIONEGRO','CARTAGO - VALLE':'CARTAGO','DISENTIS - SUIZA':'DISENTIS','PIEDECHINCHE - VALLE':'PIEDECHINCHE','BÉLGICA':'BELGICA','CALI VALLE':'CALI','BERLÃ\x8dN':'BERLIN','ZELL AM SE - AUSTRIA':'ZELL AM SE','REP.CHECA':'REPUBLICA CHECA','ANNECY - FRANCIA':'ANNECY','LIGA DE SQUASH DE BOLIVAR':'BOLIVAR','HAVANA, CUBA':'LA HAVANA','ANDRADAS - BRASIL':'ANDRADAS','PAIPA - BOYACA':'PAIPA','PAIPA - BOYACA':'PAIPA','LIGA DE SQUASH DE SANTANDER':'SANTANDER','MEDELLIN ANTIOQUIA':'MEDELLIN', 'SANTIAGO DE CHILE, CHILE': 'SANTIAGO DE CHILE','GUAYAQUIL, ECUADOR':'GUAYAQUIL','MEDELLIN - ANTIOQUIA':'MEDELLIN','LA ROMANA, REPÊBLICA DOMINICANA':'LA ROMANA','PILAR - BUENOS AIRES':'PILAR-BUENOS AIRES','PRADO TOLIMA':'PRADO','VILLAVICENCIO META':'VILLAVICENCIO','CIUDAD DE MÉXICO':'CIUDAD DE MEXICO','CIUDAD DE MÉXICO,':'CIUDAD DE MEXICO','SAN JOSE, COSTA RICA':'SAN JOSE','CUMANA, VENEZUELA':'CUMANA','CALI/BOGOTA':'CALI-BOGOTA','BOGOTA/NORTE DE SANTANDER/SANTANDER/MEDELLIN/CALI/PEREIRA/BARRANQUILLA':'BOGOTA-NORTE DE SANTANDER-SANTANDER-MEDELLIN-CALI-PEREIRA-BARRANQUILLA','GACHALA - CUNDINAMARCA':'GACHALA','EE.UU.':'EEUU','BOTOTILLO - ECUADOR':'BOTOTILLO','SACATEPEQUEZ, GUATEMALA':'SACATEPEQUEZ','COSTA NORTE, URABA, MEDELLIN, CALI':'COSTA NORTE-URABA-MEDELLIN-CALI','ANSERMA - VALLE':'ANSERMA','CASTRIES, STA. LUCIA':'CASTRIES','ZAMORA, ECUADOR':'ZAMORA','MACANAL - BOYACA':'MACANAL','BOGOTÃ\x81':'BOGOTA','USA':'EEUU','MONTERÃ\x8dA':'MONTERIA','SANTA MARTHA':'SANTA MARTA','PACHAVITA - BOYACA':'PACHAVITA','SANTA MARTA, COLOMBIA':'SANTA MARTA','LIGA DE  SQUASH DE CUNDINAMARCA':'CUNDINAMARCA','MANAGUA, NICARAGUA':'MANAGUA','LIGA DE SQUASH DE BOGOTÀ- NOGAL':'BOGOTA','VILLAVICENCIO - META':'VILLAVICENCIO','BELLO HORIZINTE -VALLE':'BELLO HORIZINTE','CALI/CUCUTA/PEREIRA/MEDELLIN/BOGOTA/BARRANQUILLA/RIOACHA/SANTANDER/':'CALI-CUCUTA-PEREIRA-MEDELLIN-BOGOTA-BARRANQUILLA-RIOACHA-SANTANDER','ROLDANILLO - VALLE':'ROLDANILLO','CORAL GABLES, FL':'CORAL GABLES','LIMA -PERU':'LIMA','PILAR-ARGENTINA':'PILAR','BOGOTÃ\x81 - COLOMBIA':'BOGOTA','TAOYUAN, CHINESE':'TAOYUAN','MEDELLÃ\x8dN':'MEDELLIN','MALAGA NORTE S/DER':'MALAGA','CHIA - CMARCA':'CHIA','TOMINÉ - MUÑA':'OTROS','CUENCA -ECUADOR':'CUENCA','SPOKANE - USA':'SPOKANE','NEUSA, CUNDINAMARCA':'NEUSA','EGYPTO/ CAIRO':'CAIRO','COLOMBIA/ PENDIENTE':'COLOMBIA','VILLETA CUNDINAMARCA':'VILLETA','BARCELONA ESPAÑA':'BARCELONA','SAN DIEGO CA USA':'SAN DIEGO','WRIGHTSVILLE BEACH/NC / USA':'WRIGHTSVILLE BEACH','NARIÑO':'NARIÑO','MONTERÃ\x8dA CORDOBA':'MONTERIA','BUENOS AIRES, ARGENTINA':'BUENOS AIRES','CALIMA - CLUB NÃ\x81UTICO DEL PACÃ\x8dFICO':'CALIMA','BUENOS AIRES (ARG)':'BUENOS AIRES','DOHA - QATAR':'DOHA','BOGOTA   COL':'BOGOTA','BUENOS AIRES - ARGENTINA':'BUENOS AIRES','LIGA CUNDINAMARCA':'CUNDINAMARCA','RIO MAGDALENA':'MAGDALENA','SOGAMOSO - BOYACÃ\x81':'SOGAMOSO','CIRCUITO EUROPEO':'OTROS','COLOMBIA/ CUNDINAMARCA':'CUNDINAMARCA','PALMIRA - VALLE - COLOMBIA':'PALMIRA','GUATAPÉ- ANTIOQUIA':'GUATAPE','MADRID - ESPAÑA':'MADRID','ST. PETERSBURG - RUSSIA':'SAN PETERSBURGO','TOCAIMA CUNDINAMARCA':'TOCAIMA','OBERHAUSEN (NORTHRHINE-WESTFALIA) - GER':'OBERHAUSEN','BOGOTA  ARRAYANES':'BOGOTA','X DEFINIR':'POR DEFINIR','MONTEVIDEO- URUGUAY':'MONTEVIDEO','SANTA MARTA MAGDALENA':'SANTA MARTA','VALLLE':'VALLE','BOGOTÃ\x81 D.C.':'BOGOTA','BOGOTá':'BOGOTA','SANTA MARTA-MAGDALENA':'SANTA MARTA','TOMINÉ - CNP':'OTROS','GIRARDOT CUNDINAMARCA':'GIRARDOT','TUNJA (BOYACA) COLOMBIA':'TUNJA','PUCÓN CHILE':'PUCANA','MIAMI- ESTADOS UNIDOS':'MIAMI','BOGOTA  (CLUB LOS LAGARTOS)':'BOGOTA','KAOSHIUNG (TAIPEI)':'KAOSHIUNG','SAN ANTONIO  TX':'SAN ANTONIO','BARRANQUILLA- CALI':'BARRANQUILLA-CALI','HANGZHOU-CHN':'HANGZHOU','CÊCUTA':'CUCUTA','OKALA FLA.':'OKALA','OVERLAND PARK KS':'OVERLAND PARK','PD':'OTROS','SACATEPÉQUEZ':'SACATEPEQUEZ','BAKÊ':'OTROS','LA ROMANA, REPÊBLICA DOMINICANA':'LA ROMANA','HAMBURG, ALEMANIA':'HAMBURG','OBERHAUSEN (NORTHRHINE-WESTFALIA) - GER':'OBERHAUSEN','COLOMBIA/ CUNDINAMARCA':'CUNDINAMARCA','NEIVA HUILA':'NEIVA','CHINA.':'CHINA','EGYPTO/ CAIRO':'CAIRO','NEUSA, CUNDINAMARCA':'NEUSA','FLORIDABLANCA , SANTANDER':'FLORIDABLANCA','SANTIAGO, REPÊBLICA DOMINICANA':'SANTIAGO','CAGUAS, PUERTO RICO':'CAGUAS','PEDAVENA - ITALIA':'PEDAVENA','SAN ANTONIO - TEXAS':'SAN ANTONIO','CALI - VALLE DEL CAUCA':'CALI','SAN ANDRE ISLAS':'SAN ANDRES','CATALUÑA':'CATALUÑA','BUFFALO  NY':'BUFFALO','LIGA DE SQUASH DE BOGOTA':'BOGOTA','PINEHURST  CAROLINA DEL NORTE':'PINEHURST-CAROLINA DEL NORTE','NEIVA - HUILA':'NEIVA','WARREN NJ':'WARREN','ZIOLKOWSK Y PYTLASINSKI':'ZIOLKOWSK-PYTLASINSKI','PALM BEACH GARDENS  FL':'PALM BEACH GARDENS','DURBANVILLE  CAPE TOWN':'DURBANVILLE-CAPE TOWN','LOMBARD ILL':'LOMBARD','PORTLAND OR':'PORTLAND','LA VIRGINIA RISARALDA':'LA VIRGINIA-RISARALDA','SAO_PAULO':'SAO PAULO','LONG BEACH CA':'LONG BEACH','CALI €“ VALLE DEL CAUCA':'CALI','OVERLAND PARKS KS':'OVERLAND','MINNEAPOLIS  MN':'MINNEAPOLIS','SAN DIEGO CA':'SAN DIEGO','SIOUX FALL SD':'SIOUX FALL','STOCKTON  CA':'STOCKTON','CINCINATTI OH':'CINCINATTI','DAVISON MI':'DAVISON','FRANCIA  ESPAÑA  ITALIA':'FRANCIA-ESPAÑA-ITALIA','WELLINGTON FLA.':'WELLINGTON','PUERTO LOPEZ META':'PUERTO LOPEZ','PERÊ':'OTROS','COLOMBIA/ PENDIENTE':'OTROS','SEATTLE WA':'SEATTLE','SÃƑO JOSÉ DOS CAMPOS':'SAN JOSE DOS CAMPOS','PEREIRA - RDA':'PEREIRA','BOGOTá':'BOGOTA','EL SALVADOR':'SAN SALVADOR','CÊCUTA':'CUCUTA','P.D':'POR DEFINIR','CORAL GABLES  FL':'CORAL GABLES','MIAMI  FL':'MIAMI','BARBOSA - SANTANDER':'BARBOSA','TARIJA,':'TARIJA','SARASOTA FL':'SARASOTA','MILLBROOK ONT':'MILLBROOK','ELOY  ARIZONA  USA':'ELOY','TORONTO  CANADA':'TORONTO','SANTA CRUZ DE LA SIERRA':'SANTA CRUZ','FOUNTAIN VALLEY CA':'FOUNTAIN VALLEY','WELLINGTON FLA':'WELLINGTON','ARLINGTON VA':'ARLINGTON','FULLERTON  CA':'FULLERTON','RESEDA CA':'RESEDA','QUITO- GUAYAQUIL':'QUITO-GUAYAQUIL','MEXICO  DF':'MEXICO','FUNZA CUND SAN ANDRES':'FUNZA-SAN ANDRES','CIUDAD DE MEXICO':'MEXICO','POR CONFIRMAR':'POR DEFINIR','STA MARTA':'SANTA MARTA'})

In [ ]:
df['LUGAR'] = df['LUGAR'].str.upper()

In [ ]:
lista = df[df['LUGAR'] == 'DEL 26 AL 30 /01/2015'].index
lista1 = df[df['LUGAR'] == 'NACIONAL'].index
lista2 = df[df['LUGAR'] == '42305'].index
lista3 = df[df['LUGAR'] == '42320'].index
lista4 = df[df['LUGAR'] == 'REGIONAL'].index
lista5 = df[df['LUGAR'] == 'ESTADOS UNIDOS'].index
lista6 = df[df['LUGAR'] == 'FEBRERO 17 AL 19'].index
lista7 = df[df['LUGAR'] == '27 JUNIO A 2 DE JULIO'].index
lista8 = df[df['LUGAR'] == 'ABRIL 29 Y 30'].index
lista9 = df[df['LUGAR'] == 'FEBRERO 17 AL 19'].index
lista10 = df[df['LUGAR'] == '27 JUNIO A 2 DE JULIO'].index
lista11 = df[df['LUGAR'] == 'JUNIO 8 A 11'].index
lista12 = df[df['LUGAR'] == 'JUNIO'].index
lista13 = df[df['LUGAR'] == '8/05/2017 0:00'].index
lista14 = df[df['LUGAR'] == 'AGOSTO 5 Y 6'].index
lista15 = df[df['LUGAR'] == 'SEPTIEMBRE'].index
lista16 = df[df['LUGAR'] == 'DICIEMBRE'].index
lista17 = df[df['LUGAR'] == 'OCTUBRE'].index
lista18 = df[df['LUGAR'] == 'NOVIEMBRE'].index
lista19 = df[df['LUGAR'] == 'NOVIEMBRE 5 Y6'].index
lista20 = df[df['LUGAR'] == 'MARZO 19 Y 20'].index 
lista21 = df[df['LUGAR'] == 'JULIO'].index 
lislugar=[lista[0],lista1[0],lista2[0],lista3[0],lista4[0],lista4[0],lista5[0],lista6[0],lista7[0],lista8[0],lista9[0],lista10[0],lista11[0],lista12[0],lista13[0],lista14[0],lista15[0],lista16[0],lista17[0],lista18[0],lista19[0],lista20[0],lista21[0]]

In [ ]:
df=df.drop(lislugar)

##Graficos y analisis

Al momento de hacer el **GROUPBY** se descartan los nulos

In [ ]:
dfd=df.reset_index()

In [ ]:
#Agrupamos las fechas y los tipos de calendarios, haciendo una cuenta de cuantos calendarios se presentan por fecha
d_f=dfd.groupby(['FECHA','TIPO DE CALENDARIO'])[['index']].count().reset_index()

**d_f**: Bodega de datos donde se presentan las fechas de calendarios, los tipos y la cantidad de calendarios, programados, en donde participan colombianos

In [ ]:
d_f

In [ ]:
d_f=d_f.reset_index()

In [ ]:
#Se agrupo por tipo de calendario y se hace una suma de todos los calendarios de este tipo
d_f=dfd.groupby(['TIPO DE CALENDARIO'])[['index']].sum().reset_index()

**¿Cuáles son los tipos de calendarios con mayor participación, por parte de los colombianos?**

In [ ]:
# crear gráfica
fig = px.pie(d_f , values = 'index', names = 'TIPO DE CALENDARIO', title = '<b>% PARTICIPACION DE CALENDARIOS SEGUN SU TIPO<b>')

# agregar detalles a la gráfica
fig.update_layout(
    template = 'plotly_dark',
    legend_title = '<b>Tipos de calendarios: <b>',
    title_x = 0.4)

fig.show()

**Análisis descriptivo**
* Se puede observar que el tipo de calendario que mas se programa, para los deportistas colombianos, segun esta base es el Deportivo con un 69,9%, seguido por calendarios de tipo Nacional con un 21,3 de las participaciones competitivas.
* Se puede observar el tipo de calendario que menos grogramaciones tiene son los de capacitacion con un 0.49%, ya que son menos regulares en comparacion de las participaciones competitivas.


In [ ]:
#Agrupamos por pais y se hizo un conteo de numero de calendarios segun el pais, ademas ordenamos de mayor a menor
ff=df.groupby(['PAIS'])[['FECHA']].count().sort_values('FECHA', ascending = False).reset_index()

In [ ]:
ff['nc'] = ff['PAIS'].str.capitalize()

**ff**: bodega de datos para almacenar los países y el número de veces que fue programado por país



In [ ]:
ff

In [ ]:
#Seleccionamos las 33 primeras filas ya que nos dimos cuenta de que habian 2 datos que incoherentes a la hora de graficar y un dato que no se podia graficar en el mapa
# que es por definir, entonces se decidio borrarlos en este punto

ff=ff.head(34)

Se buscaron los códigos de tres letras para cada país 

In [ ]:
#Borramos los datos incoherentes, y el por definir ya que podria presentar inconsistencias a la hora de graficar
l1 = ff[ff['nc'] == 'Otros'].index
l2 = ff[ff['nc'] == 'Mayores'].index 
l3 = ff[ff['nc'] == 'Por definir'].index

li=[l1[0],l2[0],l3[0]]

In [ ]:
#Borramos colombia ya que buscamos Los paises donde hubieran participado colombianos, por esta razón no se incluyo 
l4 = ff[ff['nc'] == 'COL'].index
l=l4[0]

In [ ]:
#Borramos los datos anteriormente mencionados 
ff=ff.drop(li)
ff=ff.drop(l) 

In [ ]:
#verificamos que si quedaran los códigos de los países bien
ff['nc'].unique()

array(['USA', 'BRA', 'MEX', 'ARG', 'PER', 'CAN', 'CHL', 'FRA', 'ECU',
       'ITA', 'ESP', 'BOL', 'RUS', 'DEU', 'VEN', 'PRY', 'POL', 'CHN',
       'GTM', 'CUB', 'PRT', 'URY', 'DOM', 'JPN', 'PRI', 'TUR', 'PAN',
       'SLV', 'GRC', 'CRI'], dtype=object)

In [ ]:
ff.reset_index()

index                  PAIS  FECHA   nc
0       1                  EEUU    132  USA
1       3                BRASIL     65  BRA
2       4                MEXICO     56  MEX
3       5             ARGENTINA     54  ARG
4       6                  PERU     52  PER
5       7                CANADA     47  CAN
6       8                 CHILE     45  CHL
7       9               FRANCIA     42  FRA
8      10               ECUADOR     36  ECU
9      11                ITALIA     31  ITA
10     12                ESPAÑA     28  ESP
11     13               BOLIVIA     27  BOL
12     14                 RUSIA     25  RUS
13     16              ALEMANIA     24  DEU
14     17             VENEZUELA     23  VEN
15     18              PARAGUAY     19  PRY
16     19               POLONIA     17  POL
17     20                 CHINA     17  CHN
18     21             GUATEMALA     16  GTM
19     22                  CUBA     16  CUB
20     23              PORTUGAL     15  PRT
21     24               URUGUAY     15  URY
22     25  REPUBLICA DOMINICANA     13  DOM
23     26                 JAPON     12  JPN
24     27           PUERTO RICO     11  PRI
25     28               TURQUIA     11  TUR
26     29                PANAMA     11  PAN
27     30           EL SALVADOR     11  SLV
28     32                GRECIA      9  GRC
29     33            COSTA RICA      9  CRI

In [ ]:
ff=ff.reset_index().drop('index',axis=1)

In [ ]:
#Se renombro fecha por que fue la variable utilizada para hacer el conteo del numero de calendarios realizados en ese país en concreto 
ff=ff.rename(columns={'FECHA':'PARTICIPACIONES'})

In [ ]:
ff

PAIS  PARTICIPACIONES   nc
0                   EEUU              132  USA
1                 BRASIL               65  BRA
2                 MEXICO               56  MEX
3              ARGENTINA               54  ARG
4                   PERU               52  PER
5                 CANADA               47  CAN
6                  CHILE               45  CHL
7                FRANCIA               42  FRA
8                ECUADOR               36  ECU
9                 ITALIA               31  ITA
10                ESPAÑA               28  ESP
11               BOLIVIA               27  BOL
12                 RUSIA               25  RUS
13              ALEMANIA               24  DEU
14             VENEZUELA               23  VEN
15              PARAGUAY               19  PRY
16               POLONIA               17  POL
17                 CHINA               17  CHN
18             GUATEMALA               16  GTM
19                  CUBA               16  CUB
20              PORTUGAL               15  PRT
21               URUGUAY               15  URY
22  REPUBLICA DOMINICANA               13  DOM
23                 JAPON               12  JPN
24           PUERTO RICO               11  PRI
25               TURQUIA               11  TUR
26                PANAMA               11  PAN
27           EL SALVADOR               11  SLV
28                GRECIA                9  GRC
29            COSTA RICA                9  CRI

In [ ]:
#Se creo una nueva columna para ubicar el codigo del pais de 3 letras para poder graficarlos en el mapa
ff['nc']=ff['nc'].replace({'Colombia':'COL','Eeuu':'USA','Brasil':'BRA','Mexico':'MEX','Argentina':'ARG','Peru':'PER','Canada':'CAN','Chile':'CHL','Francia':'FRA','Ecuador':'ECU','Italia':'ITA','España':'ESP','Bolivia':'BOL','Rusia':'RUS','Alemania':'DEU','Venezuela':'VEN','Paraguay':'PRY','China':'CHN','Polonia':'POL','Guatemala':'GTM','Cuba':'CUB','Uruguay':'URY','Portugal':'PRT','Republica dominicana':'DOM','Japon':'JPN','Turquia':'TUR','Puerto rico':'PRI','Panama':'PAN','El salvador':'SLV','Costa rica':'CRI','Grecia':'GRC'})

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



**¿Cuales fueron los 30 paises en donde se programaron mayor numero de calendarios, donde participaban colombianos?**   
 


In [ ]:

fig = px.scatter_geo(ff, locations="nc", color="PARTICIPACIONES", hover_name="PAIS", size="PARTICIPACIONES",
                      template = 'plotly_dark',
                      projection="natural earth",
                      title = '<b>PARTICIPACIONES INTERNACIONALES DE COLOMBIANOS<b>'
                  )
fig.show()

**Análisis descriptivo**
* Se puede observar que el pais en donde mas calendarios, se programan para la participacion de colombianos es Estados Unidos con 132 participaciones, seguido por Brazil con 65 participaciones 
* Se puede observar que el continente donde colombianos tienen mayor participacion es America
- En centro América tiene 143 participaciones
- En sur América 336 participaciones
- En norte América 179
* Podemos concluir, que la mayoria de participaciones se concentran en sur America
* Se puede concluir que en el continente que menos tienen menos calendarios programados, por lo tanto, menos participación es Asia


In [ ]:
#Agrupamos por tipo de deporte, y los ordenamos de mayor a menor
ft=df.groupby(['DEPORTE'])[['FECHA']].count().sort_values('FECHA', ascending = False).reset_index()

In [ ]:
#Se renombro fecha por que con esta hicimos el conteo de numero de calendarios donde fueron programados colombianos
ft=ft.rename(columns={'FECHA':'PARTICIPACIONES'})

**ft**: bodega de datos para almacenar los deportes y la cantidad de calendarios programados por cada deporte

In [ ]:
ft

DEPORTE  PARTICIPACIONES
0                       GOLF              163
1                      TENIS              156
2              AUTOMOVILISMO              152
3                   PATINAJE              119
4                   ECUESTRE              118
5                   GIMNASIA              105
6                      LUCHA               82
7             TIRO DEPORTIVO               72
8   ACTIVIDADES SUBACUATICAS               70
9                       JUDO               69
10                   AJEDREZ               67
11                  NATACION               63
12              MOTOCICLISMO               51
13                    BRIDGE               50
14                     SAMBO               48
15                BALONCESTO               47
16                     BOXEO               46
17                  VOLEIBOL               44
18                  ARQUERIA               40
19           DEPORTES AEREOS               37
20                      VELA               37
21                   BOWLING               34
22                    SQUASH               33
23             TENIS DE MESA               32
24                   ESGRIMA               32
25                    KARATE               31
26               MOTONAUTICA               28
27                     WUSHU               28
28                 JIU-JITSU               27
29                  CANOTAJE               27
30           FUTBOL DE SALON               26
31                     RUGBY               25
32                 ATLETISMO               24
33             ESQUI NAUTICO               24
34                      TEJO               24
35                   SOFTBOL               21
36                  JIUJITSU               18
37                     ESQUI               16
38                     KARTS               16
39                   BEISBOL               15
40           BAILE DEPORTIVO               15
41                 TAEKWONDO               13
42                   HAPKIDO               12
43                    BILLAR                5
44                      SURF                4
45                 BALONMAMO                3
46                  TRIATLON                3
47                 BADMINTON                3
48              LEV DE PESAS                3
49                 KARATE DO                2

In [ ]:
#Se seleccionaron los 10 deportes con mayor participación
ft=ft.head(10)

In [ ]:
ft

DEPORTE  PARTICIPACIONES
0                      GOLF              163
1                     TENIS              156
2             AUTOMOVILISMO              152
3                  PATINAJE              119
4                  ECUESTRE              118
5                  GIMNASIA              105
6                     LUCHA               82
7            TIRO DEPORTIVO               72
8  ACTIVIDADES SUBACUATICAS               70
9                      JUDO               69

**¿Cuales fueron los 10 deportes con mayor numero de calendarios programados para los colombianos?**

Para representar los 10 deportes se decidio usar un grafico de barras, ya que se hace de mas facil la interpretacion de los datos, y este es ideal para representar variables cualitativas.

In [ ]:
# crear gráfica
fig = px.bar(ft, x='DEPORTE', y='PARTICIPACIONES', title ='<b>PARTICIPACIONES POR DEPORTE<b>',color='DEPORTE')

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'DEPORTE',
    yaxis_title = 'PARTICIPACIONES',
    
    template = 'plotly_dark',
    title_x = 0.5)
                

fig.show()

**Análisis descriptivo**
* Se puede observar los calendarios programados por deporte, y su numero de calendarios, que estan nombrados en el eje de las Y como PARTICIPACIONES
* Se puede observar que los deportes que tienen mayor numero de calendarios programados dentro de esta base son Golf, Tenis y Automovilismo, los cuales representan un 42,85% de los datos de este ranking.
* Se puede observar que no aparece el futbol, que es uno de los deportes mas practicados en Colombia y esto se da por que no es tomado en cuenta en la base de datos, de donde se tomaron los mismos. 


# **CONSOLIDADO ESCENARIOS**

##Carga de datos

In [ ]:

import pandas as pd
import numpy as np
import plotly.express as px
from plotly.subplots import make_subplots

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
df= pd.read_csv("/content/drive/MyDrive/Analítica/Consolidado_Escenarios.csv")
df.head(5)

codigo       tipo                                          nombre  \
0     530      LIGAS                         LIGA DE LUCHA DE NARIÑO   
1    2403  ESCENARIO       CANCHA DE BALONCESTO DEL BARRIO EL JARDIN   
2    2223  ESCENARIO  PISCINA DE LA LIGA DE ACTIVIDADES SUBACUATICAS   
3    2320  ESCENARIO     CANCHA DE BALONCESTO DEL BARRIO CAMINO REAL   
4     611      LIGAS                   LIGA DE CICLISMO DE SANTANDER   

                  deporte                             direccion   telefono  \
0                   LUCHA                       KR 29A N° 18-09  7311849.0   
1  Recreativo y Deportivo                      CL 29 A BIS N?29  5566409.0   
2               Deportivo                         CL 5B N?34-51  5575833.0   
3  Recreativo y Deportivo                     CL 11 CON KR 50 A  5566409.0   
4                CICLISMO  CL 14 KR 33 VELODROMO ALFONSO FLOREZ        NaN   

     municipio     departamento                     paginaweb  \
0        PASTO           NARIÑO                           NaN   
1         CALI  VALLE DEL CAUCA    oscar.guerrero@cali.gov.co   
2         CALI  VALLE DEL CAUCA  secretarialivas@livas.com.co   
3         CALI  VALLE DEL CAUCA    oscar.guerrero@cali.gov.co   
4  BUCARAMANGA        SANTANDER        licisander@hotmail.com   

                          email           ubicacion  \
0                           NaN                 NaN   
1    oscar.guerrero@cali.gov.co                Otro   
2  secretarialivas@livas.com.co  Complejo deportivo   
3    oscar.guerrero@cali.gov.co                Otro   
4        licisander@hotmail.com                 NaN   

                                         razonsocial claseubicacion  \
0                                                NaN            NaN   
1  SECRETARIA DE RECREACION Y DEPORTE DE SANTIAGO...    Zona Urbana   
2  LIGA DE ACTIVIDADES SUBACUATICAS DEL VALLE DEL...    Zona Urbana   
3  SECRETARIA DE RECREACION Y DEPORTE DE SANTIAGO...    Zona Urbana   
4                                                NaN            NaN   

     localidadcomuna           descripcion   latitud   longitud  \
0                NaN                   NaN  1.218323 -77.280142   
1        COMUNA ONCE  CANCHA DE BALONCESTO  3.427835 -76.515341   
2  COMUNA DIECINUEVE               PISCINA  3.431317 -76.541545   
3  COMUNA DIECISIETE  CANCHA DE BALONCESTO  3.408229 -76.537650   
4                NaN                   NaN  7.134491 -73.115723   

                                           urlimagen  
0                                                NaN  
1  http://www.visorgeocoldeportes.gov.co/images/e...  
2  http://www.visorgeocoldeportes.gov.co/images/e...  
3  http://www.visorgeocoldeportes.gov.co/images/e...  
4                                                NaN

## Depuracion

In [ ]:
#Con .drop eliminamos las columnas que consideramos innecesarias, no las sobreescribimos por motivos de errores y poder retomar la original
df0= df.drop(["urlimagen","email","codigo","direccion","paginaweb","telefono"], axis=1)
df0

tipo                                          nombre  \
0         LIGAS                         LIGA DE LUCHA DE NARIÑO   
1     ESCENARIO       CANCHA DE BALONCESTO DEL BARRIO EL JARDIN   
2     ESCENARIO  PISCINA DE LA LIGA DE ACTIVIDADES SUBACUATICAS   
3     ESCENARIO     CANCHA DE BALONCESTO DEL BARRIO CAMINO REAL   
4         LIGAS                   LIGA DE CICLISMO DE SANTANDER   
...         ...                                             ...   
2487  ESCENARIO  PISTA ATLETICA DEL ESTADIO MANUEL MURILLO TORO   
2488      INDER                                INDEPORTES CAUCA   
2489  ESCENARIO                CANCHA MULTIPLE N° 1 COLOMBIATON   
2490      LIGAS         LIGA DE TAEKWONDO DE NORTE DE SANTANDER   
2491  ESCENARIO   SALA DE BOXEO N° 2 COLISEO BERNARDO CARABALLO   

                     deporte            municipio        departamento  \
0                      LUCHA                PASTO              NARIÑO   
1     Recreativo y Deportivo                 CALI     VALLE DEL CAUCA   
2                  Deportivo                 CALI     VALLE DEL CAUCA   
3     Recreativo y Deportivo                 CALI     VALLE DEL CAUCA   
4                   CICLISMO          BUCARAMANGA           SANTANDER   
...                      ...                  ...                 ...   
2487               Deportivo               IBAGUÉ              TOLIMA   
2488                     NaN              POPAYÁN               CAUCA   
2489               Deportivo  CARTAGENA DE INDIAS             BOLIVAR   
2490               TAEKWONDO               CÚCUTA  NORTE DE SANTANDER   
2491               Deportivo  CARTAGENA DE INDIAS             BOLIVAR   

               ubicacion                                        razonsocial  \
0                    NaN                                                NaN   
1                   Otro  SECRETARIA DE RECREACION Y DEPORTE DE SANTIAGO...   
2     Complejo deportivo  LIGA DE ACTIVIDADES SUBACUATICAS DEL VALLE DEL...   
3                   Otro  SECRETARIA DE RECREACION Y DEPORTE DE SANTIAGO...   
4                    NaN                                                NaN   
...                  ...                                                ...   
2487  Complejo deportivo                                     GESTORA URBANA   
2488                 NaN                                                NaN   
2489                Otro  INSTITUTO DISTRITAL DE DEPORTE Y RECREACION DE...   
2490                 NaN                                                NaN   
2491                Otro  INSTITUTO DISTRITAL DE DEPORTE Y RECREACION DE...   

     claseubicacion                localidadcomuna           descripcion  \
0               NaN                            NaN                   NaN   
1       Zona Urbana                    COMUNA ONCE  CANCHA DE BALONCESTO   
2       Zona Urbana              COMUNA DIECINUEVE               PISCINA   
3       Zona Urbana              COMUNA DIECISIETE  CANCHA DE BALONCESTO   
4               NaN                            NaN                   NaN   
...             ...                            ...                   ...   
2487    Zona Urbana                    COMUNA DIEZ    PISTA DE ATLETISMO   
2488            NaN                            NaN                   NaN   
2489    Zona Urbana  UNIDAD COMUNERA DE GOBIERNO 6       CANCHA MULTIPLE   
2490            NaN                            NaN                   NaN   
2491    Zona Urbana  UNIDAD COMUNERA DE GOBIERNO 2         SALA DE BOXEO   

        latitud   longitud  
0      1.218323 -77.280142  
1      3.427835 -76.515341  
2      3.431317 -76.541545  
3      3.408229 -76.537650  
4      7.134491 -73.115723  
...         ...        ...  
2487   4.430058 -75.217911  
2488   2.439276 -76.607872  
2489  10.416449 -75.447105  
2490   7.893488 -72.501368  
2491  10.425213 -75.536615  

[2492 rows x 12 columns]

In [ ]:
#Renombramos todas las columnas ya que algunas no estan muy claras además de que quedan mejor en mayúsculas y con espaciado 
df0= df0.rename(columns={"tipo":"CLASIFICACIÓN","nombre":"NOMBRE DEL ESCENARIO","deporte":"DEPORTE","municipio":"MUNICIPIO","departamento":"DEPARTAMENTO","ubicacion":"UBICACIÓN","razonsocial":"RAZÓN SOCIAL","claseubicacion":"TIPO ZONA","localidadcomuna":"COMUNA","descripcion":"TIPO DE ESCENARIO","latitud":"LATITUD","longitud":"LONGITUD"})
df0

In [ ]:
#De aquí para abajo empezamos a analizar cada columna en busca de datos mal escritos y su posbli corrección
#Primero aplicamos .unique para mirar que contiene
df0["DEPORTE"].unique()

array(['LUCHA', 'Recreativo y Deportivo', 'Deportivo', 'CICLISMO',
       'Recreativo', 'FUTBOL', 'TAEKWONDO', 'ATLETISMO', 'KARATE DO', nan,
       'GIMNASIA', 'BEISBOL', 'Recreativo y Educación Fisíca',
       'ACTIVIDADES SUBACUATICAS', 'PARAPENTE', 'TRIATHLON',
       'FUTBOL DE SALON', 'SOFTBOL', 'BALONMANO', 'SAMBO', 'BILLAR',
       'MOTOCICLISMO', 'TENIS DE MESA', 'RAQUETBALL', 'RUGBY',
       'FISICULTURISMO', 'BALONCESTO', 'TEJO', 'PATINAJE', 'JUDO',
       'VOLEIBOL', 'COLEO', 'NATACION', 'TIRO Y CAZA', 'WUSHU',
       'RACQUETBALL', 'AJEDREZ', 'JIUJITSU', 'SQUASH', 'AUTOMOVILISMO',
       'BOXEO', 'TIRO CON ARCO', 'TENIS', 'ESGRIMA', 'BOLOS', 'VELA',
       'PARACAIDISMO', 'LEVANTAMIENTO DE PESAS', 'CANOTAJE Y REMO',
       'ALA DELTA', 'AEROMODELISMO', 'ESQUI NAUTICO', 'BADMINTON',
       'EQUITACION', 'ECUESTRE', 'DEPORTE ESPECIAL', 'GOLF', 'KARTS',
       'PARAMOTOR', 'MOTONAUTICA', 'PELOTA A MANO CHAZA', 'HAPKIDO',
       'PLANEADORES', 'DEPORTES AEREOS', 'BOWLING', 'TE

In [ ]:
df0["DEPORTE"]=df0["DEPORTE"].replace({"Recreativo":"Recreativo y Deportivo","Deportivo":"Recreativo y Deportivo","Recreativo y Educación Fisíca":"Recreativo y Deportivo","FISICOCULTURISMO":"FISICULTURISMO","TENIS DE CAMPO":"TENIS","ARQUEROS DE COLOMBIA":"TIRO Y CAZA"})

In [ ]:
#Decidimos cambiar esta clasificación de ubicación porque era poco diciente
df0["UBICACIÓN"]=df0["UBICACIÓN"].replace({"Caja de compensaci?":"Otro","Universidad o centro de educaci? superior":"Universidad"})

In [ ]:
df0["TIPO ZONA"]=df0["TIPO ZONA"].replace({"nan":"Zona Rural"})

In [ ]:
df0["TIPO DE ESCENARIO"]=df0["TIPO DE ESCENARIO"].replace({"CAMPO DE TENIS":"CANCHA DE TENIS","SALA DE TENIS":"CANCHA DE TENIS","CANCHA DE TENIS DE CAMPO":"CANCHA DE TENIS","CANCHA DE TENIS N? 2":"CANCHA DE TENIS","CANCHA DE TENIS N? 1":"CANCHA DE TENIS","CANCHA DE MULTIPLE":"CANCHA MULTIPLE",
    "CANCHA DE MICROFUTBOL (FUTBOL DE SALON)":"CANCHA DE MICROFUTBOL","CANCHA DE MICROFUTBOL N? 2":"CANCHA DE MICROFUTBOL","CANCHA DE VOLEIBOL COLISEO":"CANCHA DE VOLEIBOL","CANCHA MICROFUTBOL":"CANCHA DE MICROFUTBOL","PISTA DE PATINAJE VELOCIDAD":"PISTA DE PATINAJE","CANCHA DE BALONCESTO PLACA CUBIERTA":"CANCHA DE BALONCESTO",
    "CANCHA DE TENIS N? 6":"CANCHA DE TENIS","CANCHS MULTIPLE":"CANCHA MULTIPLE","CANCHA VOLEIBOL":"CANCHA DE VOLEIBOL","POLIDEPORTIVO":"CANCHA MULTIPLE","COLISEO MAYOR":"CANCHA MULTIPLE","CANCHA SINTETICA DE FUTBOL 8":"CANCHA FUTBOL 8","CANCHA MULTIPLE N? 2":"CANCHA MULTIPLE","CANCHA MULTIPLE N? 1":"CANCHA MULTIPLE","CANCHA DE FUTBOL N? 2":"CANCHA DE FUTBOL",
    "CANCHA DE FUTBOL N? 1":"CANCHA DE FUTBOL","CANCHA DE TENIS POLVO DE LADRILLO":"CANCHA DE TENIS","CANCHA DE TENIS N? 7":"CANCHA DE TENIS","CANCHA DE MICRIFUTBOL":"CANCHA DE MICROFUTBOL","CANCHA MULTIPLE (MEDIA ESTRUCTURA)":"CANCHA MULTIPLE","CANCHA DE TENIS N? 4":"CANCHA DE TENIS","PISTAS PARA LA PRACTICA DE PATINAJE":"PISTA DE PATINAJE","CANCHA DE VOLEIBOL CUBIERTA":"CANCHA DE VOLEIBOL","CANCHA DE BALONCESTO COLISEO":"CANCHA DE BALONCESTO",
    "COLISEO CUBIERTO":"COLISEO","CANCHA FUTBOL 8":"CANCHA DE FUTBOL","CANCHA DE TENIS N? 3":"CANCHA DE TENIS","CANCHA DE TENIS N? 5":"CANCHA DE TENIS","PATINODROMO":"PISTA DE PATINAJE","SALAS PARA LA PRACTICA DE TAEKWONDO":"SALON DE TAEKWONDO","CANCHA DE MICROFUTBOL FOTO 030":"CANCHA DE MICROFUTBOL","CANCHA MULTIPLE PLACA CUBIERTA":"CANCHA MULTIPLE","CANCHA MULTIPLE N?1":"CANCHA MULTIPLE",
    "COLISEO CUBIERTO SAN JORGE":"COLISEO","CANCHA DE VOLEIBOL PLACA CUBIERTA":"CANCHA DE VOLEIBOL","CANCHA AUXILIAR DE MICROFUTBOL":"CANCHA DE MICROFUTBOL","CANCHA SINTETICA DE MICROFUTBOL":"CANCHA DE MICROFUTBOL","CANCHA AUXILIAR DE VOLEIBOL":"CANCHA DE VOLEIBOL", 'ESTADIO MENOR DE BEISBOL ""JOSE VARGAS"':"DIAMANTE DE BEISBOL","SALON DE BILLAR Y TENIS DE MESA":"SALA DE TENIS DE MESA","CANCHA MULTIPLE CUBIERTA":"CANCHA MULTIPLE","CANCHA DE TENIS INFANTIL":"CANCHA DE TENIS","SALON DE TENIS":"CANCHA DE TENIS",
    "CANCHA MICROFUTBOL PLACA CUBIERTA":"CANCHA DE MICROFUTBOL","SALON DE TENIS DE MESA":"SALA DE TENIS DE MESA","DIAMANTE DE SOFTBOL":"CAMPO DE SOFTBOL","SALAS O SALONES PARA LA PRACTICA DE GIMNASIA":"SALON DE GIMNASIA OLIMPICA","PISTA DE ATLETISMO EN ASFALTO":"PISTA DE ATLETISMO","COLISEO DE GIMNASIA":"SALON DE GIMNASIA OLIMPICA","SALON DE GIMNASIA RITMICA":"SALON DE GIMNASIA OLIMPICA","CAMPO DE BEISBOL":"CANCHA DE BEISBOL","SALON DE SQUASH":"CANCHA DE SQUASH",
    "COLISEO DE JUDO":"SALON DE JUDO","DIAMANTE DE BEISBOL":"CANCHA DE BEISBOL","COLISEO DE BOLOS":"SALON DE BOLOS","SALA DE TAEKWONDO":"SALON DE TAEKWONDO","CAMPO PARA PESCA":"DEPORTES DE MAR","PISTA  DE TROTE":"PISTA DE TROTE","ESTADIO DE FUTBOL":"CANCHA DE FUTBOL","PISTADE BICICROSS":"PISTA DE BICICROSS","GIMNASIO DE PESAS":"GIMNASIO","GIMNASIOS PARA LA PRACTICA DE CULTURA FISICA":"GIMNASIO","CANCHA MUTIPLE":"CANCHA MULTIPLE","SALON DE LEVANTAMIENTO DE PESAS":"GIMNASIO",
    "CANCHA DE SOFTBOL":"CAMPO DE SOFTBOL","CAMPO DE HOCKEY":"CANCHA DE HOCKEY","PISCINAINFANTIL":"PISCINA INFANTIL","SALONES DE AJEDREZ":"SALA DE AJEDREZ","TENIS DE MESA":"SALA DE TENIS DE MESA","CANCHA MULTPLE":"CANCHA MULTIPLE"})

##Graficos y analisis

In [ ]:
#Realizamos un nuevo Dataframe donde solo utilizaramos los datos necesarios, en este caso, DEPARTAMENTO, LONGITUD y LATITUD
df_d=df0[["DEPARTAMENTO","LONGITUD","LATITUD","NOMBRE DEL ESCENARIO"]]

In [ ]:
#Utilizamos estas líneas de código para los permisos de Maps
token_map = "pk.eyJ1Ijoic2Vsb3IyNyIsImEiOiJjbDdxdGVzOTIwOG9vM3VudjF2a3p3d3ZvIn0.50TpwKXSl45fMvLuVe6vxQ"
px.set_mapbox_access_token(token_map)

In [ ]:
#Generamos el map con nuestro dataframe creado
df2 = df_d.dropna(subset =['DEPARTAMENTO']) # quitar valores nulos de COMUNA
px.scatter_mapbox(df2, lat = 'LATITUD', lon = 'LONGITUD', color ='DEPARTAMENTO',template = 'plotly_dark',   mapbox_style  = 'streets',
                  color_continuous_scale = px.colors.cyclical.IceFire, size_max = 30, zoom = 10)

In [ ]:
#Decidimos crear un contador de escenarios deportivos por cada departamento para que quedara mucho más clara la información que proporcionamos en el mapa
df_d2 = df_d.groupby(['DEPARTAMENTO'])[['NOMBRE DEL ESCENARIO']].count().reset_index().sort_values('NOMBRE DEL ESCENARIO',ascending=False)
df_d2

In [ ]:
df_d2 = df_d.groupby(['DEPARTAMENTO'])[['NOMBRE DEL ESCENARIO']].count().reset_index().sort_values('NOMBRE DEL ESCENARIO',ascending=False)

dic = {'SAN ANDRES PROVIDENCIA Y SANTA CATALINA':'SA, PROV Y SC',
       'NORTE DE SANTANDER':'N.SANTANDER',
       }

df_d2['DEPARTAMENTO'] = df_d2['DEPARTAMENTO'].replace(dic)

# crear gráfica
fig = px.bar(df_d2, x='DEPARTAMENTO', y='NOMBRE DEL ESCENARIO', color = 'DEPARTAMENTO', title ='<b>Escenarios Deportivos por Departamento<b>')

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Departamentos',
    yaxis_title = 'Números de escenarios',
    template = 'plotly_dark',
    title_x = 0.5,
    legend_title = '<b>Departamento<b>')

fig.show()

El Valle del Cauca marca una considerable diferencia respecto a los otros departamentos ya que tiene la cantidad mas grande de escenarios deportivos, lo que nos permite pensar que hay una desigualdad respecto al resto del pais, ya que si miramos el mapa se ibserva que hay demasiadas zonas del pais donde no hay presencia de ningun tipo de escenario deportivo, habría que realizar grandes desplazamientos para muchos pueblos y municipios para hacer uso de estos. Esto incentiva mas nuestro analisis en la desigualdad para la población colombiana en el acceso de escenarios

**Información adicional**

10 escenarios deportivos de Cali que demuestran que el deporte es nuestra vida
Gran parte de la infraestructura deportiva de Cali nació a partir de los VI Juegos Panamericanos, celebrados en nuestra ciudad en 1971.

Este hito activó el deporte, la cultura y la economía al recibir deportistas y turistas de diferentes países. Así mismo, desencadenó una transformación urbanística de Cali con la construcción de distintos espacios para practicar deportes (desde el fútbol hasta el billar).

Por otra parte, otros escenarios deportivos de Cali fueron inaugurados a raíz de los Juegos Mundiales en 2013. Este es el caso, por ejemplo, del Coliseo de Hockey en Línea Miguel Calero junto al cual se encuentra el monumento CALI Ciudad Deportiva.

Poniendo en constraste esto con la información arrojada en las graficas se justifica la imponencia del Valle del Cauca en escenarios deportivos, y es que acá es donde se han hecho muchos eventos deportivos. 

https://constructoramelendez.com/blog/escenarios-deportivos-cali/


**Pregunta 2**: ¿Qué tipo de zona es en la que mas predominan los escenarios deportivos?

In [ ]:
#Acá hicimos el cambio de los datos nulos por Zona Rural
df_3=df0[["TIPO ZONA","DEPARTAMENTO"]].fillna("Zona Rural")

In [ ]:
#Optamos por utilizar 
base = df_3.groupby(['TIPO ZONA'])[['DEPARTAMENTO']].count().reset_index()

# crear gráfica
fig = px.pie(base , values = 'DEPARTAMENTO', names = 'TIPO ZONA', title = '<b>% Escenarios deportivos por zona<b>')

# agregar detalles a la gráfica
fig.update_layout(
    template = 'plotly_dark',
    legend_title = 'Tipo de Zona',
    title_x = 0.5)

fig.show()

Como podemos observar hay mayor porcentaje de escenarios deportivos en las zonas urbanas con un 69.1%, lo que nos hace reflexionar respecto a la importancia de la poblacion en zonas rurales y sus necesidades también de acceder a estos espacios

**Información adicional**

Cuatrocientos mil metros cuadrados en 165 escenarios deportivos o recreativos por un valor cercano a los $773 mil millones. Para este fin, en este cuatrienio, el Ministerio del Deporte aportó cerca del 92% de los recursos para la ejecución o adecuación de estas obras.
De los 165 escenarios financiados en este gobierno, 40 se ejecutan en municipios que hacen parte de los Programas de Desarrollo con Enfoque Territorial (PDET), 83 en municipios con mayor riesgo de desplazamiento y/o reclutamiento forzado de niños, niñas y adolescentes y 52 beneficiarán municipios con población perteneciente a grupos étnicos (comunidades negras, afrodescendientes, raizales o palenqueras).

Poniendo esta información en contraste con la pregunta nos da a entender que se estan haciendo esfuerzos para incrementar los escenarios en las zonas rurales o mas vulnerables del pais, o sea que se espera segun esta noticia que en los proximos años este porcentaje se equilibre.
https://www.eltiempo.com/deportes/otros-deportes/un-nuevo-pais-con-escenarios-deportivos-y-recreativos-guillermo-herrera-681290

**Pregunta 3**: ¿Qué deporte es el que más escenarios tiene? y ¿cual o cuales son los que menos tienen?

In [ ]:
#Generación de base de datos para abordar la pregunta
df_4=df0[["DEPORTE","DEPARTAMENTO"]]
df_4

DEPORTE        DEPARTAMENTO
0                      LUCHA              NARIÑO
1     Recreativo y Deportivo     VALLE DEL CAUCA
2     Recreativo y Deportivo     VALLE DEL CAUCA
3     Recreativo y Deportivo     VALLE DEL CAUCA
4                   CICLISMO           SANTANDER
...                      ...                 ...
2487  Recreativo y Deportivo              TOLIMA
2488                     NaN               CAUCA
2489  Recreativo y Deportivo             BOLIVAR
2490               TAEKWONDO  NORTE DE SANTANDER
2491  Recreativo y Deportivo             BOLIVAR

[2492 rows x 2 columns]

In [ ]:
#Groupby de deporte para el contador
df_4 = df_4.groupby(['DEPORTE'])[['DEPARTAMENTO']].count().reset_index().sort_values('DEPARTAMENTO',ascending=False)

In [ ]:
#Eliminamos la categoria de Recreación y deportivo porque al no estar en una categoria exacta preferimos no usarla para el analisis
df_5= df_4.drop([49],axis=0)
df_5

DEPORTE  DEPARTAMENTO
25             FUTBOL            68
16           CICLISMO            29
53          TAEKWONDO            28
4           ATLETISMO            28
43           PATINAJE            27
..                ...           ...
41          PARAMOTOR             1
28  GLOBO AEROSTATICO             1
38        MOTONAUTICA             1
29               GOLF             1
20           ECUESTRE             1

[62 rows x 2 columns]

In [ ]:
#Generación del groupby
df_d5 = df_5.groupby(['DEPORTE'])[['DEPARTAMENTO']].count().reset_index().sort_values('DEPARTAMENTO',ascending=False)

# crear gráfica
fig = px.bar(df_5, x='DEPORTE', y='DEPARTAMENTO', color = 'DEPORTE',  title ='<b>Cantidad de escenarios por deporte<b>')

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Deportes',
    yaxis_title = 'Numero de escenarios',
    template = 'plotly_dark',
    title_x = 0.5,
    legend_title = '<b>Deporte<b>')

fig.show()

Se nota que hay un dominio muy marcado de futbol respecto a otros deportes en escenarios deportivos en el pais, lo que nos hace cuestionar hacia donde esta dirigido el deporte en general porque hay muchos como golf, esquinautico, racketball entre otras que no tiene suficientes escenarios para el pais.

# **ESCUELAS DEPORTIVAS**

##Carga de datos

In [1]:
# importar paquetes
import pandas as pd
import numpy as np
import plotly.express as px

In [2]:
#Importar Datos
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
#Leer base de datos
df = pd.read_csv('/content/drive/MyDrive/Programa_Escuelas_Deportivas_Para_Todos.csv', sep = ",")

##Depuracion

In [4]:
#Homologacion categorias de la variable Departamento
df['Departamento']= df['Departamento'].replace(['AMAZONAS'],'Amazonas')
df['Departamento']= df['Departamento'].replace(['ANTIOQUIA'],'Antioquia')
df['Departamento']= df['Departamento'].replace(['ARAUCA'],'Arauca')
df['Departamento']= df['Departamento'].replace(['ATLÁNTICO','ATLANTICO'],'Atlántico')
df['Departamento']= df['Departamento'].replace(['BOLÍVAR','BOLIVAR'],'Bolívar')
df['Departamento']= df['Departamento'].replace(['BOYACÁ'],'Boyacá')
df['Departamento']= df['Departamento'].replace(['CALDAS'],'Caldas')
df['Departamento']= df['Departamento'].replace(['CAQUETA'],'Caquetá')
df['Departamento']= df['Departamento'].replace(['CASANARE'],'Casanare')
df['Departamento']= df['Departamento'].replace(['CAUCA'],'Cauca')
df['Departamento']= df['Departamento'].replace(['CESAR'],'Cesar')
df['Departamento']= df['Departamento'].replace(['CHOCÓ','CHOCO'],'Chocó')
df['Departamento']= df['Departamento'].replace(['CÓRDOBA','CORDOBA'],'Córdoba')
df['Departamento']= df['Departamento'].replace(['CUNDINAMARCA'],'Cundinamarca')
df['Departamento']= df['Departamento'].replace(['Guanía','GUAINÍA','GUAINIA'],'Guainía')
df['Departamento']= df['Departamento'].replace(['GUAVIARE'],'Guaviare')
df['Departamento']= df['Departamento'].replace(['HUILA'],'Huila')
df['Departamento']= df['Departamento'].replace(['LA GUAJIRA'],'La Guajira')
df['Departamento']= df['Departamento'].replace(['META'],'Meta')
df['Departamento']= df['Departamento'].replace(['NARIÑO'],'Nariño')
df['Departamento']= df['Departamento'].replace(['NORTE DE SANTANDER','Norte de santander'],'Norte de Santander')
df['Departamento']= df['Departamento'].replace(['PUTUMAYO','putumayo'],'Putumayo')
df['Departamento']= df['Departamento'].replace(['QUINDIO'],'Quindío')
df['Departamento']= df['Departamento'].replace(['SANTANDER'],'Santander')
df['Departamento']= df['Departamento'].replace(['SUCRE'],'Sucre')
df['Departamento']= df['Departamento'].replace(['TOLIMA','tolima'],'Tolima')
df['Departamento']= df['Departamento'].replace(['VALLE DEL CAUCA'],'Valle del Cauca')
df['Departamento']= df['Departamento'].replace(['VAUPÉS'],'Vaupés')
df['Departamento']= df['Departamento'].replace(['VICHADA'],'Vichada')
df['Departamento']= df['Departamento'].replace(['San Andrés , Providencia'],'San Andrés')

In [5]:
#Homologacion categorias variable Municipio
df['Municipio']= df['Municipio'].replace(['LETICIA','leticia'],'Leticia')
df['Municipio']= df['Municipio'].replace(['puerto nariño','PUERTO NARIÑO'],'Puerto Nariño')
df['Municipio']= df['Municipio'].replace(['APARTADÓ', 'APARTADO', 'Apartado'],'Apartadó')
df['Municipio']= df['Municipio'].replace(['CAMPAMENTO'],'Campamento')
df['Municipio']= df['Municipio'].replace(['CAREPA'],'Carepa') 
df['Municipio']= df['Municipio'].replace(['CHIGORODO'], 'Chigorodó')
df['Municipio']= df['Municipio'].replace(['EL BAGRE'],'El bagre')
df['Municipio']= df['Municipio'].replace(['SABANALARGA','sabanalarga'],'Sabanalarga')
df['Municipio']= df['Municipio'].replace(['san carlos'],'San Carlos')
df['Municipio']= df['Municipio'].replace(['SAN PEDRO', 'san pedro','San pedro'],'San Pedro')
df['Municipio']= df['Municipio'].replace(['SANTA BARBARA','santa barbara'],'Santa Bárbara')
df['Municipio']= df['Municipio'].replace(['ZARAGOZA','zaragoza'],'Zaragoza')
df['Municipio']= df['Municipio'].replace(['ARAUCA'],'Arauca')
df['Municipio']= df['Municipio'].replace(['ARAUQUITA','arauquita'],'Arauquita')
df['Municipio']= df['Municipio'].replace(['CRAVO NORTE'],'Cravo Norte')
df['Municipio']= df['Municipio'].replace(['FORTUL','fortul'],'Fortul')
df['Municipio']= df['Municipio'].replace(['PUERTO RONDÓN'],'Puerto Rondón')
df['Municipio']= df['Municipio'].replace(['SARAVENA'],'Saravena')
df['Municipio']= df['Municipio'].replace(['PALMAR DE VARELA','palmar de varela'],'Palmar de Varela')
df['Municipio']= df['Municipio'].replace(['REPELON','Repelon','REPELÓN'],'Repelón')
df['Municipio']= df['Municipio'].replace(['SABANAGRANDE'],'Sabanagrande')
df['Municipio']= df['Municipio'].replace(['SANTA LUCÍA','Santa lucia','SANTA LUCIA'],'Santa Lucia')
df['Municipio']= df['Municipio'].replace(['SUAN','suan'],'Suan')
df['Municipio']= df['Municipio'].replace(['TUBARÀ','TUBARA','TUBARÁ'],'Tubará')
df['Municipio']= df['Municipio'].replace(['CALAMAR'],'Calamar')
df['Municipio']= df['Municipio'].replace(['CARMEN DE BOLIVAR','EL CARMEN DE BOLIVAR'],'El carmen de bolivar')
df['Municipio']= df['Municipio'].replace(['el guamo','EL GUAMO','GUAMO'],'El Guamo')
df['Municipio']= df['Municipio'].replace(['MARÍA LA BAJA','MARIA LA BAJA'],'Maria la baja')
df['Municipio']= df['Municipio'].replace(['SAN JACINTO'],'San Jacinto Bolivar')
df['Municipio']= df['Municipio'].replace(['SAN JACINTO DEL CAUCA'],'San Jacinto del Cauca')
df['Municipio']= df['Municipio'].replace(['SAN JUAN NEPOMUCENO'],'San Juan Nepomuceno')
df['Municipio']= df['Municipio'].replace(['MOMPOX','MomPOX','mompox'],'Mompox')
df['Municipio']= df['Municipio'].replace(['VILLANUEVA','villanueva'],'Villanueva')
df['Municipio']= df['Municipio'].replace(['ZAMBRANO','zambrano'],'Zambrano')
df['Municipio']= df['Municipio'].replace(['BUENAVISTA','buenavista'],'Buenavista')
df['Municipio']= df['Municipio'].replace(['CERINZA','cerinza'],'Cerinza')
df['Municipio']= df['Municipio'].replace(['CÓMBITA'],'Cómbita')
df['Municipio']= df['Municipio'].replace(['MARIPI','maripi'],'Maripi')
df['Municipio']= df['Municipio'].replace(['MIRAFLORES'],'Miraflores')
df['Municipio']= df['Municipio'].replace(['LA PAZ','la paz','La paz','la Paz'],'La Paz')
df['Municipio']= df['Municipio'].replace(['SOTAQUIRA'],'Sotaquira')
df['Municipio']= df['Municipio'].replace(['TIBASOSA','tibasosa'],'Tibasosa')
df['Municipio']= df['Municipio'].replace(['SAN JOSÉ DEL GUAVIARE','SAN JOSÉ','San Jose'],'San Jose del Guaviare')
df['Municipio']= df['Municipio'].replace(['ALBANIA','albania'],'Albania')
df['Municipio']= df['Municipio'].replace(['BELEN DE LOS ANDAQUIES','BELÉN DE LOS ANDAQUÍES'],'Belén de los Andaquíes')
df['Municipio']= df['Municipio'].replace(['cartagena del chaira','CARTAGENA DEL CHAIRÁ'],'Cartagena del Chairá')
df['Municipio']= df['Municipio'].replace(['PUERTO RICO'],'Puerto Rico')
df['Municipio']= df['Municipio'].replace(['LA SALINA'],'La Salina')
df['Municipio']= df['Municipio'].replace(['NUNCHIA','Nunchia'],'Nunchía')
df['Municipio']= df['Municipio'].replace(['OROCUE - RESGUARDO INDIGENA SAN JUANITO','OROCUE'],'Orocué')
df['Municipio']= df['Municipio'].replace(['SAN LUIS DE PALENQUE'],'San Luis de Palenque')
df['Municipio']= df['Municipio'].replace(['TRINIDAD','trinidad'],'Trinidad')
df['Municipio']= df['Municipio'].replace(['guachene','GUACHENE','Guachene'],'Guachené')
df['Municipio']= df['Municipio'].replace(['puerto tejada','Puerto tejada','PUERTO TEJADA'],'Puerto Tejada')
df['Municipio']= df['Municipio'].replace(['santander de quilichao','SANTANDER DE QUILICHAO','santander de Quilichao','Santander de Q'],'Santander de Quilichao')
df['Municipio']= df['Municipio'].replace(['Piendamo'],'Piendamó')
df['Municipio']= df['Municipio'].replace(['POPAYAN','POPAYÁN','Popayan'],'Popayán')
df['Municipio']= df['Municipio'].replace(['LA VEGA','la vega'],'La Vega')
df['Municipio']= df['Municipio'].replace(['MERCADERES'],'Mercaderes')
df['Municipio']= df['Municipio'].replace(['INZA'],'Inzá')
df['Municipio']= df['Municipio'].replace(['JAMBALÓ','Jambalò'],'Jambaló')
df['Municipio']= df['Municipio'].replace(['CHIMICHAGUA','chimichagua'],'Chimichagua')
df['Municipio']= df['Municipio'].replace(['san alberto','SAN ALBERTO'],'San Alberto')
df['Municipio']= df['Municipio'].replace(['San Martin','san martin','SAN MARTIN','SAN MARTÍN'],'San Martín')
df['Municipio']= df['Municipio'].replace(['PELAYA','pelaya'],'Pelaya')
df['Municipio']= df['Municipio'].replace(['GAMARRA','gamarra'],'Gamarra')
df['Municipio']= df['Municipio'].replace(['manaure','MANAURE'],'Manaure')
df['Municipio']= df['Municipio'].replace(['TAMALAMEQUE','tamalameque'],'Tamalameque')
df['Municipio']= df['Municipio'].replace(['SAN DIEGO','san diego'],'San Diego')
df['Municipio']= df['Municipio'].replace(['LA GLORIA','la gloria'],'La Gloria')
df['Municipio']= df['Municipio'].replace(['ACANDI'],'Acandí')
df['Municipio']= df['Municipio'].replace(['ALTO BAUDÓ','ALTO BAUDO'],'Alto Baudó')
df['Municipio']= df['Municipio'].replace(['ATRATO YUTO','ATRATO'],'Atrato')
df['Municipio']= df['Municipio'].replace(['BAHÍA SOLANO'],'Bahía solano')  
df['Municipio']= df['Municipio'].replace(['BOJAYA'],'BOJAYÁ')   
df['Municipio']= df['Municipio'].replace(['CERTEGUI'],'Certegui')
df['Municipio']= df['Municipio'].replace(['CARMEN DEL DARIEN'],'Carmen del Darien')
df['Municipio']= df['Municipio'].replace(['El Carmen de Atrato Choco','Carmen del Carmen','El Carmen de Atrato Choco l','CARMEN DE ATRATO'],'El Carmen de Atrato')
df['Municipio']= df['Municipio'].replace(['ISTMINA','istmina'],'Istmina')
df['Municipio']= df['Municipio'].replace(['MEDIO BAUDO'],'MEDIO BAUDÓ')
df['Municipio']= df['Municipio'].replace(['MEDIO SANJUAN','M. SAN JUAN'],'MEDIO SAN JUAN')
df['Municipio']= df['Municipio'].replace(['NOVITA'],'Novita')
df['Municipio']= df['Municipio'].replace(['Nuqui','nuqui'],'NUQUÍ')
df['Municipio']= df['Municipio'].replace(['QUIBDÓ','QUIBDÓ - TAGACHI','QUIBDO','Quibdo','quibdò'],'Quibdó')
df['Municipio']= df['Municipio'].replace(['SAN JOSÉ DEL PALMAR','SAN JOSE DEL PALMAR'],'San Jose del Palmar')
df['Municipio']= df['Municipio'].replace(['sipi','Sipi'],'Sipí')
df['Municipio']= df['Municipio'].replace(['UNGUIA','UNGUÍA'],'Unguía')
df['Municipio']= df['Municipio'].replace(['CANALETE','canalete'],'Canalete') 
df['Municipio']= df['Municipio'].replace(['CHINU'],'Chinú')
df['Municipio']= df['Municipio'].replace(['cotorra','COTORRA'],'Cotorra')
df['Municipio']= df['Municipio'].replace(['LOS CÓRDOBAS','LOS CORDOBAS','LOS  CORDOBAS'],'Los Córdobas')
df['Municipio']= df['Municipio'].replace(['MOÑITOS','moñitos'],'Moñitos')
df['Municipio']= df['Municipio'].replace(['puerto escondido','Puerto escondido'],'Puerto Escondido')
df['Municipio']= df['Municipio'].replace(['PURISIMA'],'Purisima')
df['Municipio']= df['Municipio'].replace(['SAN ANTERO','San antero'],'San Antero')
df['Municipio']= df['Municipio'].replace(['SAN PELAYO'],'San Pelayo')
df['Municipio']= df['Municipio'].replace(['LORICA','lorica'],'Lorica')
df['Municipio']= df['Municipio'].replace(['SESQUILE','sesquile'],'Sesquile')
df['Municipio']= df['Municipio'].replace(['GUATAQUI'],'Guataqui')
df['Municipio']= df['Municipio'].replace(['SUPATÁ','supata'],'Supata')
df['Municipio']= df['Municipio'].replace(['UBALA'],'Ubala')
df['Municipio']= df['Municipio'].replace(['SAN JUAN DE RIOSECO','san de Rioseco'],'San Juan de Rioseco')
df['Municipio']= df['Municipio'].replace(['CHIPAQUE'],'Chipaque')
df['Municipio']= df['Municipio'].replace(['fosca','FOSCA'],'Fosca')
df['Municipio']= df['Municipio'].replace(['La palma'],'La Palma')
df['Municipio']= df['Municipio'].replace(['zipacon'],'Zipacon')
df['Municipio']= df['Municipio'].replace(['SAN BERNARDO'],'San Bernardo')
df['Municipio']= df['Municipio'].replace(['San Antonio Del Tequendama','san antonio del tequendama','San Antonio Del tequendama','san Antonio Del Tequendama','san Antonio del Tequendama'],'San Antonio del Tequendama')
df['Municipio']= df['Municipio'].replace(['INIRIDA'],'Inirida')
df['Municipio']= df['Municipio'].replace(['EL RETORNO','El RETORNO','RETORNO','Retorno'],'El Retorno')
df['Municipio']= df['Municipio'].replace(['algeciras','ALGECIRAS'],'Algeciras')
df['Municipio']= df['Municipio'].replace(['campoalegre','CAMPOALEGRE'],'Campoalegre')
df['Municipio']= df['Municipio'].replace(['tello','TELLO'],'Tello')
df['Municipio']= df['Municipio'].replace(['AGRADO','agrado'],'Agrado')
df['Municipio']= df['Municipio'].replace(['GIGANTE'],'Gigante')
df['Municipio']= df['Municipio'].replace(['SUAZA','suaza'],'Suaza')
df['Municipio']= df['Municipio'].replace(['LA ARGENTINA'],'La Argentina')
df['Municipio']= df['Municipio'].replace(['LA PLATA'],'La Plata')
df['Municipio']= df['Municipio'].replace(['NATAGA'],'Nataga')
df['Municipio']= df['Municipio'].replace(['TESALIA'],'Tesalia')
df['Municipio']= df['Municipio'].replace(['BARRANCAS'],'Barrancas')
df['Municipio']= df['Municipio'].replace(['DISTRACCION'],'Distraccion')
df['Municipio']= df['Municipio'].replace(['FONSECA','fonseca'],'Fonseca')
df['Municipio']= df['Municipio'].replace(['HATONUEVO','Hato Nuevo','HATO NUEVO'],'Hatonuevo')
df['Municipio']= df['Municipio'].replace(['URUMITA'],'Urumita')
df['Municipio']= df['Municipio'].replace(['ARACATACA'],'Aracataca')
df['Municipio']= df['Municipio'].replace(['EL BANCO'],'El Banco')
df['Municipio']= df['Municipio'].replace(['GUAMAL','guamal'],'Guamal')
df['Municipio']= df['Municipio'].replace(['PUEBLO VIEJO','pueblo viejo'],'Pueblo Viejo')
df['Municipio']= df['Municipio'].replace(['CABUYARO','cabuyaro'],'Cabuyaro')
df['Municipio']= df['Municipio'].replace(['CUBARRAL'],'Cubarral')
df['Municipio']= df['Municipio'].replace(['El dorado','EL DORADO','el dorado'],'El Dorado')
df['Municipio']= df['Municipio'].replace(['MALAMBO','malambo'],'Malambo')
df['Municipio']= df['Municipio'].replace(['GUAMAL','guamal'],'Guamal')
df['Municipio']= df['Municipio'].replace(['la macarena','LA MACARENA'],'La Macarena')
df['Municipio']= df['Municipio'].replace(['MESETAS','mesetas'],'Mesetas')
df['Municipio']= df['Municipio'].replace(['RESTREPO','restrepo'],'Restrepo')
df['Municipio']= df['Municipio'].replace(['SAN JUANITO'],'San Juanito')


In [6]:
#Homologacion categorias variable Municipio
df['Municipio']= df['Municipio'].replace(['barbacoas','BARBACOAS'],'Barbacoas')
df['Municipio']= df['Municipio'].replace(['PIZARRO','FRANCISCO PIZARRO','pizarro'],'Pizarro')
df['Municipio']= df['Municipio'].replace(['MAGÜÍ - Payán','MAGUI PAYAN'],'Magui Payan')
df['Municipio']= df['Municipio'].replace(['TUMACO','tumaco'],'Tumaco')
df['Municipio']= df['Municipio'].replace(['UNIÓNPANAMERICANA','UNION PANAMERICANA'],'Unión Panamericana')
df['Municipio']= df['Municipio'].replace(['FLORIDA'],'Florida')
df['Municipio']= df['Municipio'].replace(['PUERTO SANTANDER','P/SNTDER','santander','puerto santander'],'Puerto Santander')
df['Municipio']= df['Municipio'].replace(['BUCARASICA'],'Bucarasica')
df['Municipio']= df['Municipio'].replace(['EL TARRA','el tarra'],'El Tarra')
df['Municipio']= df['Municipio'].replace(['SARDINATA'],'Sardinata')
df['Municipio']= df['Municipio'].replace(['tibu','TIBÚ'],'Tibú')
df['Municipio']= df['Municipio'].replace(['ABREGO','ÁBREGO'],'Ábrego')
df['Municipio']= df['Municipio'].replace(['CONVENCIÓN','CONVENCION'],'Convención')
df['Municipio']= df['Municipio'].replace(['EL CARMEN'],'El Carmen')
df['Municipio']= df['Municipio'].replace(['HACARÍ','hacari'],'Hacarí')
df['Municipio']= df['Municipio'].replace(['GUACARÍ','GUACARI'],'Guacarí')
df['Municipio']= df['Municipio'].replace(['SAN CALIXTO','san calixto'],'San Calixto')
df['Municipio']= df['Municipio'].replace(['TEORAMA'],'Teorama')
df['Municipio']= df['Municipio'].replace(['SANTIAGO-COLON','COLON PTYO'],'Santiago')
df['Municipio']= df['Municipio'].replace(['MOCOA','mocoa'],'Mocoa')
df['Municipio']= df['Municipio'].replace(['ORITO'],'Orito')
df['Municipio']= df['Municipio'].replace(['PUERTO ASIS'],'Puerto Asís')
df['Municipio']= df['Municipio'].replace(['PTO CAICEDO'],'Puerto Caicedo')
df['Municipio']= df['Municipio'].replace(['PUERTO GUZMAN'],'Puerto Guzmán')
df['Municipio']= df['Municipio'].replace(['SAN FRANCISCO-SIBUNDOY'],'San Francisco')
df['Municipio']= df['Municipio'].replace(['VALLE DEL GUAMUEZ-SAN MIGUEL'],'San Miguel')
df['Municipio']= df['Municipio'].replace(['VALLE DEL GUAMUEZ-SAN MIGUEL','VALLE DEL GUAMUEZ'],'Valle Del Guamuez')
df['Municipio']= df['Municipio'].replace(['VILLAGARZON','Villagarzon'],'Villagarzón')
df['Municipio']= df['Municipio'].replace(['circasia','CIRCASIA'],'Circasia')
df['Municipio']= df['Municipio'].replace(['FILANDIA'],'Filandia')
df['Municipio']= df['Municipio'].replace(['GENOVA'],'Genova')
df['Municipio']= df['Municipio'].replace(['MONTENEGRO','montenegro'],'Montenegro')
df['Municipio']= df['Municipio'].replace(['QUIMBAYA'],'Quimbaya')
df['Municipio']= df['Municipio'].replace(['la virginia'],'La Virginia')
df['Municipio']= df['Municipio'].replace(['marsella','MARSELLA'],'Marsella')
df['Municipio']= df['Municipio'].replace(['sibundoy','SIBUNDOY'],'Sibundoy')
df['Municipio']= df['Municipio'].replace(['CURITI','curiti'],'Curiti')
df['Municipio']= df['Municipio'].replace(['LOS SANTOS','Los santos','los Santos'],'Los Santos')
df['Municipio']= df['Municipio'].replace(['MOLAGAVITA'],'Molagavita')
df['Municipio']= df['Municipio'].replace(['OIBA','oiba'],'Oiba')
df['Municipio']= df['Municipio'].replace(['PALMAR','palmar'],'Palmar')
df['Municipio']= df['Municipio'].replace(['paramo','PARAMO'],'Paramo')
df['Municipio']= df['Municipio'].replace(['SABANA DE TORRES'],'Sabana de torres')
df['Municipio']= df['Municipio'].replace(['San gil','SAN GIL'],'San Gil')
df['Municipio']= df['Municipio'].replace(['San Jose de Suaita','SUAITA'],'Suaita')
df['Municipio']= df['Municipio'].replace(['chalan','CHALAN'],'Chalan')
df['Municipio']= df['Municipio'].replace(['MORROA'],'Morroa')
df['Municipio']= df['Municipio'].replace(['OVEJAS','ovejas'],'Ovejas')
df['Municipio']= df['Municipio'].replace(['coveñas','COVEÑAS'],'Coveñas')
df['Municipio']= df['Municipio'].replace(['San antonio de palmito','San Antonio de palmito','San Antonio de Palmito','PALMITO'],'Palmito')
df['Municipio']= df['Municipio'].replace(['ONOFRE'],'San Onofre')
df['Municipio']= df['Municipio'].replace(['TOLUVIEJO','toluviejo'],'Toluviejo')
df['Municipio']= df['Municipio'].replace(['COROZAL','corozal'],'Corozal')
df['Municipio']= df['Municipio'].replace(['LOS PALMITOS'],'Los Palmitos')
df['Municipio']= df['Municipio'].replace(['Sincé'],'Since')
df['Municipio']= df['Municipio'].replace(['HERVEO'],'Herveo')
df['Municipio']= df['Municipio'].replace(['SANTA ISABEL'],'Santa Isabel')
df['Municipio']= df['Municipio'].replace(['ataco','ATACO'],'Ataco')
df['Municipio']= df['Municipio'].replace(['chaparral','CHAPARRAL'],'Chaparral')
df['Municipio']= df['Municipio'].replace(['ALCALÁ','alcala','ALCALA'],'Alcalá')
df['Municipio']= df['Municipio'].replace(['BUENAVENTURA','buenaventura'],'Buenaventura')
df['Municipio']= df['Municipio'].replace(['CAICEDONIA'],'Caicedonia')
df['Municipio']= df['Municipio'].replace(['CANDELARIA'],'Candelaria')
df['Municipio']= df['Municipio'].replace(['ANTIOQUIA'],'Antioquia')
df['Municipio']= df['Municipio'].replace(['GINEBRA','ginebra'],'Ginebra')
df['Municipio']= df['Municipio'].replace(['VERSALLES'],'Versalles')
df['Municipio']= df['Municipio'].replace(['CARURU','caruru'],'Caruru')
df['Municipio']= df['Municipio'].replace(['MITU'],'Mitú')
df['Municipio']= df['Municipio'].replace(['TARAIRA'],'Taraira')
df['Municipio']= df['Municipio'].replace(['CUMARIBO'],'Cumaribo')
df['Municipio']= df['Municipio'].replace(['La primavera'],'La Primavera')
df['Municipio']= df['Municipio'].replace(['PUERTO CARREÑO'],'Puerto Carreño')
df['Municipio']= df['Municipio'].replace(['SANTA ROSALIA'],'Santa Rosalía')
df['Municipio']= df['Municipio'].replace(['SAN ANTONIO DE PADUA'],'VIGIA DEL FUERTE')
df['Municipio']= df['Municipio'].replace(['CHOCO'],'Choco')
df['Municipio']= df['Municipio'].replace(['la meta'],'meta')
df['Municipio']= df['Municipio'].replace(['BARRIO MARIA ANTONIA'],'NECOCLI')
df['Municipio']= df['Municipio'].replace(['BOLÍVAR'],'Bolivar')
df['Municipio']= df['Municipio'].replace(['S.J URE'],'San Jose De Ure')



In [7]:
#Homologación categorías de la variable Deporte 1
df['Deporte 1'] = df['Deporte 1'].replace(['BÉISBOL','BEISBOL'], 'Beisbol')
df['Deporte 1'] = df['Deporte 1'].replace(['FÚTBOL DE SALÓN','FÚTBOL SALA','Fútbol de salon','Fútbol de salón','FUTBOL DE SALON','Futbol de Salon', 'fútbol de salón','futbol de salon','FÚTBOL SE SALÓN','Futbol de salón','FÚTBOL SALÓN','FUTBOL SALON','FUTBOL DE SALÓN','FÙTBOL DE SALÒN'], 'Fútbol de Salón')
df['Deporte 1'] = df['Deporte 1'].replace(['FUTBOL','FÚTBOL','Futbol','futbol','Fulbol'], 'Fútbol')
df['Deporte 1'] = df['Deporte 1'].replace(['PATINAJE','patinaje'], 'Patinaje')
df['Deporte 1'] = df['Deporte 1'].replace(['BOXEO'], 'Boxeo')
df['Deporte 1'] = df['Deporte 1'].replace(['VOLEIBOL','Voleibol piso'],'Voleibol')
df['Deporte 1'] = df['Deporte 1'].replace(['ATLETISMO'], 'Atletismo')  
df['Deporte 1'] = df['Deporte 1'].replace(['BALONCESTO', 'baloncesto'],'Baloncesto')
df['Deporte 1'] = df['Deporte 1'].replace(['AJEDREZ'],'Ajedrez')    
df['Deporte 1'] = df['Deporte 1'].replace(['RUGBY'],'Rugby')
df['Deporte 1'] = df['Deporte 1'].replace(['TAEKWONDO', 'taekwondo'], 'Taekwondo')
df['Deporte 1'] = df['Deporte 1'].replace(['KARATE DO', 'KARATE'],'Karate')
df['Deporte 1'] = df['Deporte 1'].replace(['JUDO', 'judo'], 'Judo')
df['Deporte 1'] = df['Deporte 1'].replace(['TIRO CON ARCO', 'Tiro Arco'], 'Tiro con arco')
df['Deporte 1'] = df['Deporte 1'].replace(['LUCHA', 'lucha'],'Lucha')
df['Deporte 1'] = df['Deporte 1'].replace(['BALONMANO', 'Balon mano'], 'Balonmano')
df['Deporte 1'] = df['Deporte 1'].replace(['NATACIÓN', 'NATACION', 'NATACIOON'],'Natación')
df['Deporte 1']= df['Deporte 1'].replace(['CICLISMO'], 'Ciclismo')
df['Deporte 1']= df['Deporte 1'].replace(['LEVANTAMIENTO DE PESAS'],'Levantamiento de pesas' )
df['Deporte 1']= df['Deporte 1'].replace(['CANOTAJE'],'Canotaje')
df['Deporte 1']= df['Deporte 1'].replace(['SURF'],'Surf')
df['Deporte 1']= df['Deporte 1'].replace(['balonpesado'],'Balonpesado')

In [8]:
#Renombrar columna 'Mes\ninscripción'
df = df.rename(columns ={'Mes\ninscripción':'Mes inscripción'}) 


In [9]:
#Hacer una copia de la BD
df2 = df.copy()

In [10]:
# Se decide tratar los nulos con ffill, ya que estos datos son ingresados cronologicamente
df3 = df2.ffill() # aplicar método
df3

N° Departamento        Municipio        Deporte 1     Deporte 2  \
0          1      Córdoba           Lorica          Beisbol  NO INSCRIBIÓ   
1          2      Córdoba           Lorica          Beisbol  NO INSCRIBIÓ   
2          3      Córdoba           Lorica          Beisbol  NO INSCRIBIÓ   
3          4      Córdoba           Lorica          Beisbol  NO INSCRIBIÓ   
4          5      Córdoba           Lorica          Beisbol  NO INSCRIBIÓ   
...      ...          ...              ...              ...           ...   
32255  32256      Córdoba  San Jose De Ure  Fútbol de Salón  NO INSCRIBIÓ   
32256  32257      Córdoba  San Jose De Ure  Fútbol de Salón  NO INSCRIBIÓ   
32257  32258      Córdoba  San Jose De Ure  Fútbol de Salón  NO INSCRIBIÓ   
32258  32259      Córdoba  San Jose De Ure  Fútbol de Salón  NO INSCRIBIÓ   
32259  32260      Córdoba  San Jose De Ure  Fútbol de Salón  NO INSCRIBIÓ   

          Deporte 3     Deporte 4  Edad     Género Grupo étnico  \
0      NO INSCRIBIÓ  NO INSCRIBIÓ  10.0  Masculino       Ningún   
1      NO INSCRIBIÓ  NO INSCRIBIÓ   9.0  Masculino       Ningún   
2      NO INSCRIBIÓ  NO INSCRIBIÓ  10.0   Femenino         Afro   
3      NO INSCRIBIÓ  NO INSCRIBIÓ  12.0  Masculino       Ningún   
4      NO INSCRIBIÓ  NO INSCRIBIÓ  10.0   Femenino       Ningún   
...             ...           ...   ...        ...          ...   
32255  NO INSCRIBIÓ  NO INSCRIBIÓ   8.0  Masculino       Ningún   
32256  NO INSCRIBIÓ  NO INSCRIBIÓ   8.0  Masculino       Ningún   
32257  NO INSCRIBIÓ  NO INSCRIBIÓ   8.0  Masculino         Afro   
32258  NO INSCRIBIÓ  NO INSCRIBIÓ   9.0  Masculino       Ningún   
32259  NO INSCRIBIÓ  NO INSCRIBIÓ   9.0  Masculino         Afro   

      Condición de discapacidad       Grupo Vulnerable  Día inscripción  \
0                       Ninguna                 Ningún              4.0   
1                       Ninguna                 Ningún              7.0   
2                       Ninguna                 Ningún              3.0   
3                       Ninguna                 Ningún             10.0   
4                       Ninguna                 Ningún              3.0   
...                         ...                    ...              ...   
32255                   Ninguna  Victima del Conflicto             16.0   
32256                   Ninguna                 Ningún             16.0   
32257                   Ninguna                 Ningún             16.0   
32258                   Ninguna                 Ningún             16.0   
32259                   Ninguna                 Ningún             16.0   

       Mes inscripción  Año inscripción  
0                  6.0            2.019  
1                  6.0            2.019  
2                  6.0            2.019  
3                  6.0            2.019  
4                  6.0            2.019  
...                ...              ...  
32255              9.0         2021.000  
32256              9.0         2021.000  
32257              9.0         2021.000  
32258              9.0         2021.000  
32259              9.0         2021.000  

[32260 rows x 15 columns]

In [11]:
#se reemplazan los años
df3['Año inscripción'] = df3['Año inscripción'].replace({2021.000: '2021',2020.000:'2020', 2.019:'2019'})

In [12]:
#se pasan estas variables a entero para poder tratarlas
df3[['Día inscripción']] = df3[['Día inscripción']].astype(int)
df3[['Mes inscripción']] = df3[['Mes inscripción']].astype(int)
df3[['Año inscripción']] = df3[['Año inscripción']].astype(int)

In [13]:
#se eliminan algunos datos con condiciones
df3 = df3.drop(df3[df3['Día inscripción']<=1].index)
df3 = df3.drop(df3[df3['Día inscripción']>=31].index)

In [14]:
#se crea un Dataframe con las columnas más relevantes
graf=df3.loc[:,['Departamento','Edad','Género','Día inscripción','Mes inscripción','Año inscripción','Deporte 1']]
graf

Departamento  Edad     Género  Día inscripción  Mes inscripción  \
0          Córdoba  10.0  Masculino                4                6   
1          Córdoba   9.0  Masculino                7                6   
2          Córdoba  10.0   Femenino                3                6   
3          Córdoba  12.0  Masculino               10                6   
4          Córdoba  10.0   Femenino                3                6   
...            ...   ...        ...              ...              ...   
32255      Córdoba   8.0  Masculino               16                9   
32256      Córdoba   8.0  Masculino               16                9   
32257      Córdoba   8.0  Masculino               16                9   
32258      Córdoba   9.0  Masculino               16                9   
32259      Córdoba   9.0  Masculino               16                9   

       Año inscripción        Deporte 1  
0                 2019          Beisbol  
1                 2019          Beisbol  
2                 2019          Beisbol  
3                 2019          Beisbol  
4                 2019          Beisbol  
...                ...              ...  
32255             2021  Fútbol de Salón  
32256             2021  Fútbol de Salón  
32257             2021  Fútbol de Salón  
32258             2021  Fútbol de Salón  
32259             2021  Fútbol de Salón  

[30788 rows x 7 columns]

In [15]:
#Convertir las fechas a objeto para poder concatenar
graf[['Día inscripción','Mes inscripción','Año inscripción']] = graf[['Día inscripción','Mes inscripción','Año inscripción']].astype(str)
graf.dtypes

Departamento        object
Edad               float64
Género              object
Día inscripción     object
Mes inscripción     object
Año inscripción     object
Deporte 1           object
dtype: object

In [16]:
#se concatena la fecha
graf['Full date'] = graf['Día inscripción'].str.cat(graf['Mes inscripción'],sep="/")
graf['Full date'] = graf['Full date'].str.cat(graf['Año inscripción'],sep="/")
graf

Departamento  Edad     Género Día inscripción Mes inscripción  \
0          Córdoba  10.0  Masculino               4               6   
1          Córdoba   9.0  Masculino               7               6   
2          Córdoba  10.0   Femenino               3               6   
3          Córdoba  12.0  Masculino              10               6   
4          Córdoba  10.0   Femenino               3               6   
...            ...   ...        ...             ...             ...   
32255      Córdoba   8.0  Masculino              16               9   
32256      Córdoba   8.0  Masculino              16               9   
32257      Córdoba   8.0  Masculino              16               9   
32258      Córdoba   9.0  Masculino              16               9   
32259      Córdoba   9.0  Masculino              16               9   

      Año inscripción        Deporte 1  Full date  
0                2019          Beisbol   4/6/2019  
1                2019          Beisbol   7/6/2019  
2                2019          Beisbol   3/6/2019  
3                2019          Beisbol  10/6/2019  
4                2019          Beisbol   3/6/2019  
...               ...              ...        ...  
32255            2021  Fútbol de Salón  16/9/2021  
32256            2021  Fútbol de Salón  16/9/2021  
32257            2021  Fútbol de Salón  16/9/2021  
32258            2021  Fútbol de Salón  16/9/2021  
32259            2021  Fútbol de Salón  16/9/2021  

[30788 rows x 8 columns]

In [17]:
#Convertir la fecha en el formato ideal
graf['FECHA']=pd.to_datetime(graf['Full date'])
type(graf['FECHA'][0])

pandas._libs.tslibs.timestamps.Timestamp

In [18]:
#Homologación de categoría género
graf['Género'] = graf['Género'].replace(['FEMENINO'], 'Femenino')

In [19]:
#Función para determinar grupo etario
def funcion(x):
    if x <= 5:
        valor = 'Primera infancia'
    elif x <= 14: 
        valor = 'Infancia'
    else:
        valor = 'Adolescencia'
    return valor

In [20]:
# Aplicamos la función Apply a la variable 'Edad' y creamos la variable Grupo Etario
graf['Grupo Etario']=graf['Edad'].apply(funcion)
graf

Departamento  Edad     Género Día inscripción Mes inscripción  \
0          Córdoba  10.0  Masculino               4               6   
1          Córdoba   9.0  Masculino               7               6   
2          Córdoba  10.0   Femenino               3               6   
3          Córdoba  12.0  Masculino              10               6   
4          Córdoba  10.0   Femenino               3               6   
...            ...   ...        ...             ...             ...   
32255      Córdoba   8.0  Masculino              16               9   
32256      Córdoba   8.0  Masculino              16               9   
32257      Córdoba   8.0  Masculino              16               9   
32258      Córdoba   9.0  Masculino              16               9   
32259      Córdoba   9.0  Masculino              16               9   

      Año inscripción        Deporte 1  Full date      FECHA Grupo Etario  
0                2019          Beisbol   4/6/2019 2019-04-06     Infancia  
1                2019          Beisbol   7/6/2019 2019-07-06     Infancia  
2                2019          Beisbol   3/6/2019 2019-03-06     Infancia  
3                2019          Beisbol  10/6/2019 2019-10-06     Infancia  
4                2019          Beisbol   3/6/2019 2019-03-06     Infancia  
...               ...              ...        ...        ...          ...  
32255            2021  Fútbol de Salón  16/9/2021 2021-09-16     Infancia  
32256            2021  Fútbol de Salón  16/9/2021 2021-09-16     Infancia  
32257            2021  Fútbol de Salón  16/9/2021 2021-09-16     Infancia  
32258            2021  Fútbol de Salón  16/9/2021 2021-09-16     Infancia  
32259            2021  Fútbol de Salón  16/9/2021 2021-09-16     Infancia  

[30788 rows x 10 columns]

##Graficos y analisis

**¿Cuál es el porcentaje de participación de los deportistas según el grupo etario al que pertenecen?**

In [21]:
##Se agrupa por grupo etario 
graf3=graf.groupby(["Grupo Etario"])[["Edad"]].count().reset_index().rename(columns={'Edad':'Total deportistas'})
graf3 

Grupo Etario  Total deportistas
0      Adolescencia               1266
1          Infancia              29408
2  Primera infancia                114

In [22]:
# crear gráfica:
fig = px.pie(graf3 , values = 'Total deportistas', names = 'Grupo Etario', title = '<b>%Porcentaje de participación según clasificación etaria<b>')

# agregar detalles a la gráfica:
fig.update_layout(
    template = 'plotly_dark',
    legend_title = 'Grupo',
    title_x = 0.5)


**Análisis:**  

Para la realización del gráfico se organizó la edad por grupo etario y se analizó el porcentaje de participación de cada grupo sobre el total de deportistas, encontrando que la mayor participación la tienen los Infantes con un 95.5% de participación sobre el total; esto se debe a que dentro de las escuelas deportivas la mayoria de estudiantes se enentran entre los 5 y los 14 años, con un promedio de 10 años.


**¿Cuál es el total de deportistas por departamento?**

In [23]:
#Se agrupa por departamento
graf2=graf.groupby(["Departamento"])[["Edad"]].count().reset_index().rename(columns={'Edad':'Total deportistas'})
graf2

Departamento  Total deportistas
0             Amazonas                341
1            Antioquia               2908
2               Arauca                629
3            Atlántico               1445
4              Bolívar               1424
5               Boyacá                575
6               Caldas                521
7              Caquetá                245
8             Casanare                955
9                Cauca               1079
10               Cesar               1115
11               Chocó               2564
12        Cundinamarca                640
13             Córdoba               1206
14             Guainía                577
15            Guaviare                634
16               Huila                797
17          La Guajira               1021
18           Magdalena                367
19                Meta                894
20              Nariño               1127
21  Norte de Santander               1121
22            Putumayo               1027
23             Quindío                799
24           Risaralda                259
25          San Andrés                 40
26           Santander               1279
27               Sucre               1465
28              Tolima               1125
29     Valle del Cauca               1796
30              Vaupés                348
31             Vichada                465

In [24]:
# crear gráfica
fig = px.bar(graf2, x='Departamento', y='Total deportistas', color = 'Departamento',title ='<b>Número de deportistas por departamento<b>')

# agregar detalles a la gráfica
fig.update_layout(
    xaxis_title = 'Departamento',
    yaxis_title = 'Total Deportistas',
    template = 'plotly_dark',
    title_x = 0.5,
    legend_title = '<b>Departamento<b>')
fig.show()

**Análisis:**  

Dentro de los departamentos que más contribuyen en la participación deportiva son Antioquia con 2908, Chocó con 2564 y Valle del Cauca con 1796, lo cual puede deberse a que son los departamentos de los que más se tienen régistros además de la influencia que tienen varios deportistas sobresalientes sobre los menores de estos departamentos, a diferencia de San Andres con 40 y Caquetá con 245, siendo los departamentos con menos participación, lo cual se debe no solo al tamaño y cantidad de registros, sino también a las oportunidades que tienen los menores allí.

**¿Qué tanto contribuye cada deporte al total de deportistas por año?**

In [25]:
#Se agrupa por Año inscripcion y deporte1
graf1=graf.groupby(["Año inscripción","Deporte 1"])[["Edad"]].count().reset_index().rename(columns={'Edad':'Total deportistas'})
graf1

Año inscripción      Deporte 1  Total deportistas
0             2019        Ajedrez                176
1             2019      Atletismo                452
2             2019     Baloncesto               1007
3             2019      Balonmano                 40
4             2019        Beisbol                187
..             ...            ...                ...
64            2021           Surf                 22
65            2021      Taekwondo                153
66            2021  Tenis de mesa                 20
67            2021  Tiro con arco                120
68            2021       Voleibol                744

[69 rows x 3 columns]

In [26]:
fig = px.area(graf1, x = 'Año inscripción', y ='Total deportistas', color = 'Deporte 1',
             title= '<b>Contribución al total de deportistas por deporte por año<b>',
             color_discrete_sequence=px.colors.qualitative.G10)

fig.update_layout(
    xaxis_title = 'Año inscripción',
    yaxis_title = 'Total deportistas',
    template = 'plotly_dark',
    title_x = 0.5)
fig.show()

**Análisis:**  

El deporte que mayor contribución tiene en todos los años es el Fútbol con un total de 3643 y el Fútbol de salón con 2312, esto puede deberse a que en Colombia el deporte más influyente es el Fútbol ya que es el que cuenta con mayor apoyo y públicidad